{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Katmalu/hse23_hw1/blob/main/KM_Methylation.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CLLeQw9Mxab4"
      },
      "source": [
        "## Ссылка на диск с картированными чтениями"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iX7NcoC_CLY0"
      },
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RQAK_lggxsXi"
      },
      "source": [
        "[Bam-файлы и отчеты](https://drive.google.com/drive/folders/1athn52a93obOwfuHynDCRPRhjAPB80aO?usp=sharing)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5l0a7C6JHe40"
      },
      "source": [
        "# Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "esYQuHKO8nAg",
        "outputId": "a05e0aa6-f5c5-46e9-8526-c332ae1b91db"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-13 12:14:45--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  5.16MB/s    in 1.9s    \n",
            "\n",
            "2023-02-13 12:14:48 (5.16 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_zEx3USYPKoR"
      },
      "source": [
        "Можно получать данные секвенировия с ENA. Например, https://www.ebi.ac.uk/ena/browser/view/SRR5836473"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c5QMGKJYxrza",
        "outputId": "0c8a9110-b644-4160-92e0-4317aaec8133"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-13 12:14:48--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => ‘SRR5836473_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq. 100%[===================>]   6.38G  31.3MB/s    in 4m 54s  \n",
            "\n",
            "2023-02-13 12:19:45 (22.2 MB/s) - ‘SRR5836473_1.fastq.gz’ saved [6856386252]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d SRR5836473_1.fastq.gz\n",
        "!./FastQC/fastqc SRR5836473_1.fastq"
      ],
      "metadata": {
        "id": "U8wQi2dfm2F8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8c4f893c-e530-420b-d3a9-b10a68adad5a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of SRR5836473_1.fastq\n",
            "Approx 5% complete for SRR5836473_1.fastq\n",
            "Approx 10% complete for SRR5836473_1.fastq\n",
            "Approx 15% complete for SRR5836473_1.fastq\n",
            "Approx 20% complete for SRR5836473_1.fastq\n",
            "Approx 25% complete for SRR5836473_1.fastq\n",
            "Approx 30% complete for SRR5836473_1.fastq\n",
            "Approx 35% complete for SRR5836473_1.fastq\n",
            "Approx 40% complete for SRR5836473_1.fastq\n",
            "Approx 45% complete for SRR5836473_1.fastq\n",
            "Approx 50% complete for SRR5836473_1.fastq\n",
            "Approx 55% complete for SRR5836473_1.fastq\n",
            "Approx 60% complete for SRR5836473_1.fastq\n",
            "Approx 65% complete for SRR5836473_1.fastq\n",
            "Approx 70% complete for SRR5836473_1.fastq\n",
            "Approx 75% complete for SRR5836473_1.fastq\n",
            "Approx 80% complete for SRR5836473_1.fastq\n",
            "Approx 85% complete for SRR5836473_1.fastq\n",
            "Approx 90% complete for SRR5836473_1.fastq\n",
            "Approx 95% complete for SRR5836473_1.fastq\n",
            "Analysis complete for SRR5836473_1.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ugk3bV0LI5El"
      },
      "source": [
        "# Установка Bismark с зависимостями (+bowtie2, samtools)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uSOZLyn5n5Nl",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3ea5767c-cba4-4d3c-8f09-2eb5855ff3ab"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "📦 Installing...\n",
            "📌 Adjusting configuration...\n",
            "🩹 Patching environment...\n",
            "⏲ Done in 0:00:20\n",
            "🔁 Restarting kernel...\n"
          ]
        }
      ],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "npb--DNtn52o",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "742f6e51-e555-4b0b-a089-37b97ed17c98"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "import condacolab\n",
        "condacolab.check()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "MfBSN4YuoCL6",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "04619944-b916-4cf8-f528-ad28c187d2cc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bismark\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bismark-0.24.0             |       hdfd78af_0         6.6 MB  bioconda\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    ca-certificates-2022.12.7  |       ha878542_0         143 KB  conda-forge\n",
            "    certifi-2022.12.7          |     pyhd8ed1ab_0         147 KB  conda-forge\n",
            "    colorama-0.4.6             |     pyhd8ed1ab_0          25 KB  conda-forge\n",
            "    conda-22.11.1              |   py38h578d9bd_1         905 KB  conda-forge\n",
            "    hisat2-2.2.1               |       h87f3376_4        16.5 MB  bioconda\n",
            "    k8-0.2.5                   |       hd03093a_2         1.7 MB  bioconda\n",
            "    libhwloc-2.8.0             |       h32351e8_1         3.0 MB  conda-forge\n",
            "    minimap2-2.24              |       h7132678_1         484 KB  bioconda\n",
            "    openssl-3.0.8              |       h0b41bf4_0         2.5 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    pluggy-1.0.0               |     pyhd8ed1ab_5          16 KB  conda-forge\n",
            "    ruamel.yaml-0.17.21        |   py38h0a891b7_2         172 KB  conda-forge\n",
            "    ruamel.yaml.clib-0.2.7     |   py38h1de0b5d_1         143 KB  conda-forge\n",
            "    samtools-1.3.1             |                0         1.5 MB  bioconda\n",
            "    tbb-2021.7.0               |       h924138e_1         1.5 MB  conda-forge\n",
            "    tqdm-4.64.1                |     pyhd8ed1ab_0          82 KB  conda-forge\n",
            "    zlib-1.2.13                |       h166bdaf_4          92 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        64.6 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bismark            bioconda/noarch::bismark-0.24.0-hdfd78af_0 None\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 None\n",
            "  colorama           conda-forge/noarch::colorama-0.4.6-pyhd8ed1ab_0 None\n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-h87f3376_4 None\n",
            "  k8                 bioconda/linux-64::k8-0.2.5-hd03093a_2 None\n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.8.0-h32351e8_1 None\n",
            "  minimap2           bioconda/linux-64::minimap2-2.24-h7132678_1 None\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 None\n",
            "  pluggy             conda-forge/noarch::pluggy-1.0.0-pyhd8ed1ab_5 None\n",
            "  ruamel.yaml        conda-forge/linux-64::ruamel.yaml-0.17.21-py38h0a891b7_2 None\n",
            "  ruamel.yaml.clib   conda-forge/linux-64::ruamel.yaml.clib-0.2.7-py38h1de0b5d_1 None\n",
            "  samtools           bioconda/linux-64::samtools-1.3.1-0 None\n",
            "  tbb                conda-forge/linux-64::tbb-2021.7.0-h924138e_1 None\n",
            "  tqdm               conda-forge/noarch::tqdm-4.64.1-pyhd8ed1ab_0 None\n",
            "  zlib               conda-forge/linux-64::zlib-1.2.13-h166bdaf_4 None\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2022.9.24-ha878542_0 --> 2022.12.7-ha878542_0 None\n",
            "  certifi                            2022.9.24-pyhd8ed1ab_0 --> 2022.12.7-pyhd8ed1ab_0 None\n",
            "  conda                               22.9.0-py38h578d9bd_2 --> 22.11.1-py38h578d9bd_1 None\n",
            "  openssl                                  3.0.7-h0b41bf4_1 --> 3.0.8-h0b41bf4_0 None\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "samtools-1.3.1       | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.78it/s]\n",
            "certifi-2022.12.7    | 147 KB    | : 100% 1.0/1 [00:00<00:00, 17.16it/s]\n",
            "pluggy-1.0.0         | 16 KB     | : 100% 1.0/1 [00:00<00:00, 17.30it/s]\n",
            "minimap2-2.24        | 484 KB    | : 100% 1.0/1 [00:00<00:00,  4.88it/s]\n",
            "ruamel.yaml-0.17.21  | 172 KB    | : 100% 1.0/1 [00:00<00:00,  7.09it/s]\n",
            "tbb-2021.7.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  9.09it/s]\n",
            "bismark-0.24.0       | 6.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.98s/it]               \n",
            "tqdm-4.64.1          | 82 KB     | : 100% 1.0/1 [00:00<00:00,  9.96it/s]\n",
            "colorama-0.4.6       | 25 KB     | : 100% 1.0/1 [00:00<00:00, 16.91it/s]\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:04<00:00,  4.42s/it]               \n",
            "openssl-3.0.8        | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  6.28it/s]\n",
            "ruamel.yaml.clib-0.2 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 21.73it/s]\n",
            "conda-22.11.1        | 905 KB    | : 100% 1.0/1 [00:00<00:00,  4.34it/s]\n",
            "zlib-1.2.13          | 92 KB     | : 100% 1.0/1 [00:00<00:00, 16.22it/s]\n",
            "perl-5.32.1          | 14.4 MB   | : 100% 1.0/1 [00:05<00:00,  5.33s/it]               \n",
            "ca-certificates-2022 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 19.52it/s]\n",
            "k8-0.2.5             | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.41it/s]\n",
            "hisat2-2.2.1         | 16.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.85s/it]               \n",
            "libhwloc-2.8.0       | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.14it/s]\n",
            "Preparing transaction: | \b\b/ \b\b- \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Retrieving notices: ...working... done\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda bismark"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lBUia_8DP-gu"
      },
      "source": [
        "Можно загрузить файл с google drive напрямую:"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install libncurses5"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "srZlhzZrlbid",
        "outputId": "01451eb0-a3d6-4ca2-e3c5-904ff471951b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  libncurses5\n",
            "0 upgraded, 1 newly installed, 0 to remove and 21 not upgraded.\n",
            "Need to get 96.9 kB of archives.\n",
            "After this operation, 330 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 libncurses5 amd64 6.2-0ubuntu2 [96.9 kB]\n",
            "Fetched 96.9 kB in 1s (193 kB/s)\n",
            "Selecting previously unselected package libncurses5:amd64.\n",
            "(Reading database ... 128048 files and directories currently installed.)\n",
            "Preparing to unpack .../libncurses5_6.2-0ubuntu2_amd64.deb ...\n",
            "Unpacking libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Setting up libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! gdown \"1A0yrDgfIyBir3fPgT9_mclEZNOUkhCfP&confirm=t\"\n",
        "! gdown \"1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B&confirm=t\"\n",
        "! gdown \"1NmzWngRAZvChd1C7JlNkIr2iQShW1BRm&confirm=t\"\n",
        "! gdown \"1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg&confirm=t\"\n",
        "! gdown \"1q74W3m0bC41q5x29yO2G4tjesLl-1jih&confirm=t\" \n",
        "! gdown \"14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40&confirm=t\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EgxRSJ5LnSRD",
        "outputId": "f740911d-56df-468f-e7a0-4c54e5b12385"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1A0yrDgfIyBir3fPgT9_mclEZNOUkhCfP&confirm=t\n",
            "To: /content/SRR5836473_1_bismark_bt2_pe.bam\n",
            "100% 667M/667M [00:05<00:00, 116MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B&confirm=t\n",
            "To: /content/SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 3.03MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1NmzWngRAZvChd1C7JlNkIr2iQShW1BRm&confirm=t\n",
            "To: /content/SRR3824222_1_bismark_bt2_pe.bam\n",
            "100% 1.56G/1.56G [00:12<00:00, 121MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg&confirm=t\n",
            "To: /content/SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 2.04MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1q74W3m0bC41q5x29yO2G4tjesLl-1jih&confirm=t\n",
            "To: /content/SRR5836475_1_bismark_bt2_pe.bam\n",
            "100% 960M/960M [00:07<00:00, 126MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40&confirm=t\n",
            "To: /content/SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "100% 1.89k/1.89k [00:00<00:00, 1.74MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools sort SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam\n",
        "! samtools sort SRR3824222_1_bismark_bt2_pe.bam -o sorted_epiblast.bam\n",
        "! samtools sort SRR5836475_1_bismark_bt2_pe.bam -o sorted_ICM.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sBku9tHLpY0z",
        "outputId": "acbea108-7ea9-498f-e79b-6d85fde63b6c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 5 files...\n",
            "[bam_sort_core] merging from 11 files...\n",
            "[bam_sort_core] merging from 7 files...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools index  sorted_8cell.bam\n",
        "! samtools index  sorted_epiblast.bam\n",
        "! samtools index  sorted_ICM.bam"
      ],
      "metadata": {
        "id": "DzoAjEiuyjhv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Lw--6KavBoXp"
      },
      "source": [
        "Сколько ридов закартировалось на регион 11347700-11367700 в образце "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b-ZhJMlax75S",
        "outputId": "086b10bf-c62c-409e-9e2a-83c8564007a5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "8cell: \n",
            "1090\n",
            "epiblast: \n",
            "2328\n",
            "ICM: \n",
            "1456\n"
          ]
        }
      ],
      "source": [
        "print(\"8cell: \")\n",
        "! samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "print(\"epiblast: \")\n",
        "! samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "print(\"ICM: \")\n",
        "! samtools view sorted_ICM.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Gs0h7J9uzp6v"
      },
      "source": [
        "Сколько ридов закартировалось на регион 40185800-40195800 в образце "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "966eced5-a28d-4d08-a7c8-9aa42b328e34",
        "id": "I43JN1TZzsgw"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "8cell: \n",
            "464\n",
            "epiblast: \n",
            "1062\n",
            "ICM: \n",
            "630\n"
          ]
        }
      ],
      "source": [
        "print(\"8cell: \")\n",
        "! samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
        "print(\"epiblast: \")\n",
        "! samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l\n",
        "print(\"ICM: \")\n",
        "! samtools view sorted_ICM.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CliRCSUQTmV-"
      },
      "source": [
        "Дедупликация"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ShW-Fhzq19kE",
        "outputId": "df84802c-dd5f-4435-ac53-2764efff4a90"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_8_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_8_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2850231\n",
            "Total number duplicated alignments removed:\t521904 (18.31%)\n",
            "Duplicated alignments were found at:\t230432 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2328327 (81.69% of total)\n",
            "\n",
            "Output filename was given as: s_epiblast\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_epiblast.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7039016\n",
            "Total number duplicated alignments removed:\t205258 (2.92%)\n",
            "Duplicated alignments were found at:\t164796 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6833758 (97.08% of total)\n",
            "\n",
            "Output filename was given as: s_ICM\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_ICM.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4159998\n",
            "Total number duplicated alignments removed:\t377882 (9.08%)\n",
            "Duplicated alignments were found at:\t229833 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3782116 (90.92% of total)\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_8_cell  SRR5836473_1_bismark_bt2_pe.bam\n",
        "! deduplicate_bismark  --bam  --paired  -o s_epiblast  SRR3824222_1_bismark_bt2_pe.bam\n",
        "! deduplicate_bismark  --bam  --paired  -o s_ICM  SRR5836475_1_bismark_bt2_pe.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mUlhNr8EZAUZ",
        "outputId": "cbdc8ea2-227b-4c7a-fc6d-d6ae4d943d99"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1Gh6YYO8zmCiUfZ4ShVcvFp3J_gXjnoOF&confirm=t\n",
            "To: /content/mouse_chr.zip\n",
            "100% 366M/366M [00:03<00:00, 107MB/s]\n"
          ]
        }
      ],
      "source": [
        "! gdown \"1Gh6YYO8zmCiUfZ4ShVcvFp3J_gXjnoOF&confirm=t\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4NZ1s3i5bUUq",
        "outputId": "ba7e2008-99c6-4512-85d0-4f500d51f9fc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  mouse_chr.zip\n",
            "   creating: mouse_chr/\n",
            "   creating: mouse_chr/Bisulfite_Genome/\n",
            "   creating: mouse_chr/Bisulfite_Genome/CT_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/genome_mfa.CT_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.3.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.1.bt2  \n",
            "   creating: mouse_chr/Bisulfite_Genome/GA_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/genome_mfa.GA_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.3.bt2  \n",
            "  inflating: mouse_chr/chr11.fa      \n"
          ]
        }
      ],
      "source": [
        "! unzip mouse_chr.zip"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sf_yqXE71QjG"
      },
      "source": [
        "Время работы: ±30 мин"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Коллинг метилирования"
      ],
      "metadata": {
        "id": "LKBD3cwo1wLx"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4c5BEvKpDw5K",
        "outputId": "c76d13a6-97ad-4b8e-d73a-27be37e1f94f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_8_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_8_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_8_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_8_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2328327 lines in total\n",
            "Total number of methylation call strings processed: 4656654\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104224209\n",
            "\n",
            "Total methylated C's in CpG context:\t2101489\n",
            "Total methylated C's in CHG context:\t269204\n",
            "Total methylated C's in CHH context:\t944695\n",
            "\n",
            "Total C to T conversions in CpG context:\t2731177\n",
            "Total C to T conversions in CHG context:\t22964928\n",
            "Total C to T conversions in CHH context:\t75212716\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.2%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_8_cell.deduplicated.txt.gz CpG_OB_s_8_cell.deduplicated.txt.gz CHG_OT_s_8_cell.deduplicated.txt.gz CHG_OB_s_8_cell.deduplicated.txt.gz CHH_OT_s_8_cell.deduplicated.txt.gz CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_8_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_8_cell.deduplicated.txt.gz\t/content/CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_8_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_8_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_8_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_8_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_8_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1736778 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n",
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_epiblast.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_epiblast.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_epiblast.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_epiblast.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "Processed lines: 4000000\n",
            "Processed lines: 4500000\n",
            "Processed lines: 5000000\n",
            "Processed lines: 5500000\n",
            "Processed lines: 6000000\n",
            "Processed lines: 6500000\n",
            "\n",
            "Processed 6833758 lines in total\n",
            "Total number of methylation call strings processed: 13667516\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t330808061\n",
            "\n",
            "Total methylated C's in CpG context:\t11544557\n",
            "Total methylated C's in CHG context:\t1572954\n",
            "Total methylated C's in CHH context:\t3695988\n",
            "\n",
            "Total C to T conversions in CpG context:\t3465254\n",
            "Total C to T conversions in CHG context:\t71995231\n",
            "Total C to T conversions in CHH context:\t238534077\n",
            "\n",
            "C methylated in CpG context:\t76.9%\n",
            "C methylated in CHG context:\t2.1%\n",
            "C methylated in CHH context:\t1.5%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_epiblast.deduplicated.txt.gz CpG_OB_s_epiblast.deduplicated.txt.gz CHG_OT_s_epiblast.deduplicated.txt.gz CHG_OB_s_epiblast.deduplicated.txt.gz CHH_OT_s_epiblast.deduplicated.txt.gz CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_epiblast.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_epiblast.deduplicated.txt.gz\t/content/CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_epiblast.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_epiblast.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_epiblast.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_epiblast.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_epiblast.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2249610 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n",
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_ICM.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_ICM.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_ICM.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_ICM.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_ICM.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "\n",
            "Processed 3782116 lines in total\n",
            "Total number of methylation call strings processed: 7564232\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t177157754\n",
            "\n",
            "Total methylated C's in CpG context:\t1914374\n",
            "Total methylated C's in CHG context:\t297421\n",
            "Total methylated C's in CHH context:\t1066806\n",
            "\n",
            "Total C to T conversions in CpG context:\t6262904\n",
            "Total C to T conversions in CHG context:\t39194423\n",
            "Total C to T conversions in CHH context:\t128421826\n",
            "\n",
            "C methylated in CpG context:\t23.4%\n",
            "C methylated in CHG context:\t0.8%\n",
            "C methylated in CHH context:\t0.8%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_ICM.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_ICM.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_ICM.deduplicated.txt.gz CpG_OB_s_ICM.deduplicated.txt.gz CHG_OT_s_ICM.deduplicated.txt.gz CHG_OB_s_ICM.deduplicated.txt.gz CHH_OT_s_ICM.deduplicated.txt.gz CHH_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_ICM.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_ICM.deduplicated.txt.gz\t/content/CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_ICM.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_ICM.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_ICM.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_ICM.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_ICM.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_ICM.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_ICM.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_ICM.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_ICM.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_ICM.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2053827 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_8_cell.deduplicated.bam\n",
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_epiblast.deduplicated.bam\n",
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_ICM.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6RQ7FwM-UsSi",
        "outputId": "cb091c5e-9bb5-4ab0-cbad-f6acb0b3a988"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "User specifified dedup report: s_8_cell.deduplication_report.txt\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> s_8_cell.deduplication_report.txt <\n",
            "Processing deduplication report s_8_cell.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> s_8_cell.deduplicated_splitting_report.txt <\n",
            "Processing splitting report s_8_cell.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> s_8_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_8_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark2report --alignment_report SRR5836473_1_bismark_bt2_PE_report.txt --dedup_report s_8_cell.deduplication_report.txt --mbias_report s_8_cell.deduplicated.M-bias.txt --splitting_report s_8_cell.deduplicated_splitting_report.txt\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report --alignment_report SRR3824222_1_bismark_bt2_PE_report.txt --dedup_report s_epiblast.deduplication_report.txt --mbias_report s_epiblast.deduplicated.M-bias.txt --splitting_report s_epiblast.deduplicated_splitting_report.txt\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aElp_uxtnuT-",
        "outputId": "51e0de26-803c-49c1-c76b-5763afb56b03"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "User specifified dedup report: s_epiblast.deduplication_report.txt\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> s_epiblast.deduplication_report.txt <\n",
            "Processing deduplication report s_epiblast.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> s_epiblast.deduplicated_splitting_report.txt <\n",
            "Processing splitting report s_epiblast.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> s_epiblast.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_epiblast.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report --alignment_report SRR5836475_1_bismark_bt2_PE_report.txt --dedup_report s_ICM.deduplication_report.txt --mbias_report s_ICM.deduplicated.M-bias.txt --splitting_report s_ICM.deduplicated_splitting_report.txt\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1b0d37d0-edcd-4bc9-da4a-63109fb22d1b",
        "id": "02pcJVM3omqw"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "User specifified dedup report: s_ICM.deduplication_report.txt\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> s_ICM.deduplication_report.txt <\n",
            "Processing deduplication report s_ICM.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following splitting report:\t\t> s_ICM.deduplicated_splitting_report.txt <\n",
            "Processing splitting report s_ICM.deduplicated_splitting_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following M-bias report:\t\t> s_ICM.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_ICM.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "b6k9REsM6NXn",
        "outputId": "7288f30b-6d61-4573-93b1-5c8dbabd173a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting pyGenomeTracks\n",
            "  Downloading pyGenomeTracks-3.8-py2.py3-none-any.whl (113 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m113.6/113.6 kB\u001b[0m \u001b[31m3.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting future>=0.17.0\n",
            "  Downloading future-0.18.3.tar.gz (840 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m840.9/840.9 kB\u001b[0m \u001b[31m9.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pyBigWig>=0.3.16\n",
            "  Downloading pyBigWig-0.3.18.tar.gz (64 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.8/64.8 kB\u001b[0m \u001b[31m6.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting matplotlib<=3.6.2,>=3.1.1\n",
            "  Downloading matplotlib-3.6.2-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (9.4 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m9.4/9.4 MB\u001b[0m \u001b[31m21.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting bx-python>=0.8.13\n",
            "  Downloading bx_python-0.9.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.8/4.8 MB\u001b[0m \u001b[31m42.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting intervaltree>=2.1.0\n",
            "  Downloading intervaltree-3.1.0.tar.gz (32 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pybedtools>=0.8.1\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m55.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pyfaidx>=0.1.3\n",
            "  Downloading pyfaidx-0.7.1.tar.gz (103 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.2/103.2 kB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting hicmatrix>=15\n",
            "  Downloading HiCMatrix-15-py3-none-any.whl (37 kB)\n",
            "Collecting gffutils>=0.9\n",
            "  Downloading gffutils-0.11.1.tar.gz (1.6 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m50.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pysam>=0.14\n",
            "  Downloading pysam-0.20.0-cp38-cp38-manylinux_2_24_x86_64.whl (16.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m16.1/16.1 MB\u001b[0m \u001b[31m37.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.8/site-packages (from pyGenomeTracks) (4.64.1)\n",
            "Collecting pytest\n",
            "  Downloading pytest-7.2.1-py3-none-any.whl (317 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m317.1/317.1 kB\u001b[0m \u001b[31m21.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting numpy>=1.20\n",
            "  Downloading numpy-1.24.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m17.3/17.3 MB\u001b[0m \u001b[31m26.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting six>=1.12.0\n",
            "  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)\n",
            "Collecting argh>=0.26.2\n",
            "  Downloading argh-0.27.2-py2.py3-none-any.whl (43 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m43.3/43.3 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting argcomplete>=1.9.4\n",
            "  Downloading argcomplete-2.0.0-py2.py3-none-any.whl (37 kB)\n",
            "Collecting simplejson\n",
            "  Downloading simplejson-3.18.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (135 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m135.5/135.5 kB\u001b[0m \u001b[31m11.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting scipy>=1.2.*\n",
            "  Downloading scipy-1.10.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m34.5/34.5 MB\u001b[0m \u001b[31m27.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting cooler>=0.8.9\n",
            "  Downloading cooler-0.9.1-py2.py3-none-any.whl (103 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.9/103.9 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pandas>=0.25.*\n",
            "  Downloading pandas-1.5.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.2/12.2 MB\u001b[0m \u001b[31m81.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting tables>=3.5.*\n",
            "  Downloading tables-3.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.5/6.5 MB\u001b[0m \u001b[31m85.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting sortedcontainers<3.0,>=2.0\n",
            "  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)\n",
            "Collecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting contourpy>=1.0.1\n",
            "  Downloading contourpy-1.0.7-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (300 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m300.0/300.0 kB\u001b[0m \u001b[31m26.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.4.0-cp38-cp38-manylinux_2_28_x86_64.whl (3.4 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.4/3.4 MB\u001b[0m \u001b[31m69.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m247.7/247.7 kB\u001b[0m \u001b[31m1.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting packaging>=20.0\n",
            "  Using cached packaging-23.0-py3-none-any.whl (42 kB)\n",
            "Collecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m98.3/98.3 kB\u001b[0m \u001b[31m9.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.4.4-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m57.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.38.0-py3-none-any.whl (965 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m965.4/965.4 kB\u001b[0m \u001b[31m52.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.8/site-packages (from pyfaidx>=0.1.3->pyGenomeTracks) (65.5.1)\n",
            "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.8/site-packages (from pytest->pyGenomeTracks) (1.0.0)\n",
            "Collecting iniconfig\n",
            "  Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)\n",
            "Collecting exceptiongroup>=1.0.0rc8\n",
            "  Downloading exceptiongroup-1.1.0-py3-none-any.whl (14 kB)\n",
            "Collecting tomli>=1.0.0\n",
            "  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Collecting attrs>=19.2.0\n",
            "  Downloading attrs-22.2.0-py3-none-any.whl (60 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m60.0/60.0 kB\u001b[0m \u001b[31m6.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting multiprocess\n",
            "  Downloading multiprocess-0.70.14-py38-none-any.whl (132 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m132.0/132.0 kB\u001b[0m \u001b[31m10.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting h5py>=2.5\n",
            "  Downloading h5py-3.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.7 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.7/4.7 MB\u001b[0m \u001b[31m74.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting cytoolz\n",
            "  Downloading cytoolz-0.12.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.8 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.8/1.8 MB\u001b[0m \u001b[31m71.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pyyaml\n",
            "  Downloading PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (701 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m701.2/701.2 kB\u001b[0m \u001b[31m48.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting asciitree\n",
            "  Downloading asciitree-0.3.3.tar.gz (4.0 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting click>=7\n",
            "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m96.6/96.6 kB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pytz>=2020.1\n",
            "  Downloading pytz-2022.7.1-py2.py3-none-any.whl (499 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m499.4/499.4 kB\u001b[0m \u001b[31m34.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting blosc2~=2.0.0\n",
            "  Downloading blosc2-2.0.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.9 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.9/3.9 MB\u001b[0m \u001b[31m78.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting py-cpuinfo\n",
            "  Downloading py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)\n",
            "Collecting numexpr>=2.6.2\n",
            "  Downloading numexpr-2.8.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (381 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m381.7/381.7 kB\u001b[0m \u001b[31m31.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting cython>=0.29.21\n",
            "  Downloading Cython-0.29.33-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m69.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting msgpack\n",
            "  Downloading msgpack-1.0.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (322 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m322.5/322.5 kB\u001b[0m \u001b[31m27.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.8/site-packages (from cytoolz->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.12.0)\n",
            "Collecting dill>=0.3.6\n",
            "  Downloading dill-0.3.6-py3-none-any.whl (110 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m110.5/110.5 kB\u001b[0m \u001b[31m12.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: future, gffutils, intervaltree, pybedtools, pyBigWig, pyfaidx, asciitree\n",
            "  Building wheel for future (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for future: filename=future-0.18.3-py3-none-any.whl size=492025 sha256=1738590629ea8435b2dc3482f1b506a055058401ab69d2ec545e9d2b063153b8\n",
            "  Stored in directory: /root/.cache/pip/wheels/a6/db/41/71a0e5d071a14e716cc11bb021a9caa8f76ec337eca071487e\n",
            "  Building wheel for gffutils (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for gffutils: filename=gffutils-0.11.1-py3-none-any.whl size=1619347 sha256=f95cf74d3f4bf011e0fb0f5bcda9007c30fd1173a91cd31e563b6c0e481de53d\n",
            "  Stored in directory: /root/.cache/pip/wheels/bf/29/9e/a1d3fbccfd9359b55b63392c5bc3b124820b79ca13b87da78c\n",
            "  Building wheel for intervaltree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervaltree: filename=intervaltree-3.1.0-py2.py3-none-any.whl size=26098 sha256=ac872085ec63bd559ac720d63604a315927c48eb04f2351e1250fc62904002df\n",
            "  Stored in directory: /root/.cache/pip/wheels/84/83/6a/98a64d52a91da0bf81a9b7830f2f0d2ec749d4eac4742087c1\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp38-cp38-linux_x86_64.whl size=12767076 sha256=f92e5d3fd24eaf0499aec08472649ed3c3b85e4fb0ca2166e01e0f32766b12a9\n",
            "  Stored in directory: /root/.cache/pip/wheels/56/da/e4/fa89b7c39ffe27585c9fdc6a4d3599a5c6d47c4e07661a2a5c\n",
            "  Building wheel for pyBigWig (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyBigWig: filename=pyBigWig-0.3.18-cp38-cp38-linux_x86_64.whl size=55794 sha256=4084d091be8409e73ba79d4c878c0385f63e8eff69eec1bc416fb62f704ef9e9\n",
            "  Stored in directory: /root/.cache/pip/wheels/04/15/6c/5d167d148e049c27e758f0ecedb8c83149cbc5dccacdcca496\n",
            "  Building wheel for pyfaidx (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyfaidx: filename=pyfaidx-0.7.1-py3-none-any.whl size=27728 sha256=9d7dec6cfa803296ebc88c45653951eecb39356ce1ad4ea152cb2cc3b358a495\n",
            "  Stored in directory: /root/.cache/pip/wheels/d4/e0/ed/7fc0afff37f6cce5b8b83e3f86c8b4d28c64ceca476d6f1d78\n",
            "  Building wheel for asciitree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for asciitree: filename=asciitree-0.3.3-py3-none-any.whl size=5032 sha256=29181d66030f796912fc1cf0f197e8634762f9058da4a700813eb25c4638ed3a\n",
            "  Stored in directory: /root/.cache/pip/wheels/57/4e/7e/2e22b4e071a608b9db564a344dd47f09b4c0f6fe0e03338b6c\n",
            "Successfully built future gffutils intervaltree pybedtools pyBigWig pyfaidx asciitree\n",
            "Installing collected packages: sortedcontainers, pytz, pysam, pyBigWig, py-cpuinfo, msgpack, asciitree, argh, tomli, six, simplejson, pyyaml, pyparsing, pillow, packaging, numpy, kiwisolver, intervaltree, iniconfig, future, fonttools, exceptiongroup, dill, cytoolz, cython, cycler, click, blosc2, attrs, argcomplete, scipy, python-dateutil, pytest, pyfaidx, pybedtools, numexpr, multiprocess, h5py, contourpy, bx-python, tables, pandas, matplotlib, gffutils, cooler, hicmatrix, pyGenomeTracks\n",
            "Successfully installed argcomplete-2.0.0 argh-0.27.2 asciitree-0.3.3 attrs-22.2.0 blosc2-2.0.0 bx-python-0.9.0 click-8.1.3 contourpy-1.0.7 cooler-0.9.1 cycler-0.11.0 cython-0.29.33 cytoolz-0.12.1 dill-0.3.6 exceptiongroup-1.1.0 fonttools-4.38.0 future-0.18.3 gffutils-0.11.1 h5py-3.8.0 hicmatrix-15 iniconfig-2.0.0 intervaltree-3.1.0 kiwisolver-1.4.4 matplotlib-3.6.2 msgpack-1.0.4 multiprocess-0.70.14 numexpr-2.8.4 numpy-1.24.2 packaging-23.0 pandas-1.5.3 pillow-9.4.0 py-cpuinfo-9.0.0 pyBigWig-0.3.18 pyGenomeTracks-3.8 pybedtools-0.9.0 pyfaidx-0.7.1 pyparsing-3.0.9 pysam-0.20.0 pytest-7.2.1 python-dateutil-2.8.2 pytz-2022.7.1 pyyaml-6.0 scipy-1.10.0 simplejson-3.18.3 six-1.16.0 sortedcontainers-2.4.0 tables-3.8.0 tomli-2.0.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver"
                ]
              }
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting ipykernel\n",
            "  Downloading ipykernel-6.21.2-py3-none-any.whl (149 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m149.7/149.7 kB\u001b[0m \u001b[31m636.5 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: packaging in /usr/local/lib/python3.8/site-packages (from ipykernel) (23.0)\n",
            "Collecting jupyter-core!=5.0.*,>=4.12\n",
            "  Downloading jupyter_core-5.2.0-py3-none-any.whl (94 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m94.3/94.3 kB\u001b[0m \u001b[31m7.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting nest-asyncio\n",
            "  Downloading nest_asyncio-1.5.6-py3-none-any.whl (5.2 kB)\n",
            "Collecting comm>=0.1.1\n",
            "  Downloading comm-0.1.2-py3-none-any.whl (6.5 kB)\n",
            "Collecting matplotlib-inline>=0.1\n",
            "  Downloading matplotlib_inline-0.1.6-py3-none-any.whl (9.4 kB)\n",
            "Collecting jupyter-client>=6.1.12\n",
            "  Downloading jupyter_client-8.0.2-py3-none-any.whl (103 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.3/103.3 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting traitlets>=5.4.0\n",
            "  Downloading traitlets-5.9.0-py3-none-any.whl (117 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m117.4/117.4 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting psutil\n",
            "  Downloading psutil-5.9.4-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (280 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m280.2/280.2 kB\u001b[0m \u001b[31m10.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting debugpy>=1.6.5\n",
            "  Downloading debugpy-1.6.6-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m14.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting ipython>=7.23.1\n",
            "  Downloading ipython-8.10.0-py3-none-any.whl (784 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m784.3/784.3 kB\u001b[0m \u001b[31m19.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pyzmq>=20\n",
            "  Downloading pyzmq-25.0.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m22.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting tornado>=6.1\n",
            "  Downloading tornado-6.2-cp37-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (423 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m424.0/424.0 kB\u001b[0m \u001b[31m22.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting stack-data\n",
            "  Downloading stack_data-0.6.2-py3-none-any.whl (24 kB)\n",
            "Collecting pickleshare\n",
            "  Downloading pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)\n",
            "Collecting jedi>=0.16\n",
            "  Downloading jedi-0.18.2-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m27.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting backcall\n",
            "  Downloading backcall-0.2.0-py2.py3-none-any.whl (11 kB)\n",
            "Collecting pygments>=2.4.0\n",
            "  Downloading Pygments-2.14.0-py3-none-any.whl (1.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m30.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting decorator\n",
            "  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)\n",
            "Collecting pexpect>4.3\n",
            "  Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m59.0/59.0 kB\u001b[0m \u001b[31m6.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting prompt-toolkit<3.1.0,>=3.0.30\n",
            "  Downloading prompt_toolkit-3.0.36-py3-none-any.whl (386 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m386.4/386.4 kB\u001b[0m \u001b[31m28.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/site-packages (from jupyter-client>=6.1.12->ipykernel) (2.8.2)\n",
            "Collecting importlib-metadata>=4.8.3\n",
            "  Downloading importlib_metadata-6.0.0-py3-none-any.whl (21 kB)\n",
            "Collecting platformdirs>=2.5\n",
            "  Downloading platformdirs-3.0.0-py3-none-any.whl (14 kB)\n",
            "Collecting zipp>=0.5\n",
            "  Downloading zipp-3.13.0-py3-none-any.whl (6.7 kB)\n",
            "Collecting parso<0.9.0,>=0.8.0\n",
            "  Downloading parso-0.8.3-py2.py3-none-any.whl (100 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m100.8/100.8 kB\u001b[0m \u001b[31m10.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting ptyprocess>=0.5\n",
            "  Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)\n",
            "Collecting wcwidth\n",
            "  Downloading wcwidth-0.2.6-py2.py3-none-any.whl (29 kB)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/site-packages (from python-dateutil>=2.8.2->jupyter-client>=6.1.12->ipykernel) (1.16.0)\n",
            "Collecting executing>=1.2.0\n",
            "  Downloading executing-1.2.0-py2.py3-none-any.whl (24 kB)\n",
            "Collecting asttokens>=2.1.0\n",
            "  Downloading asttokens-2.2.1-py2.py3-none-any.whl (26 kB)\n",
            "Collecting pure-eval\n",
            "  Downloading pure_eval-0.2.2-py3-none-any.whl (11 kB)\n",
            "Installing collected packages: wcwidth, pure-eval, ptyprocess, pickleshare, executing, backcall, zipp, traitlets, tornado, pyzmq, pygments, psutil, prompt-toolkit, platformdirs, pexpect, parso, nest-asyncio, decorator, debugpy, asttokens, stack-data, matplotlib-inline, jupyter-core, jedi, importlib-metadata, comm, jupyter-client, ipython, ipykernel\n",
            "Successfully installed asttokens-2.2.1 backcall-0.2.0 comm-0.1.2 debugpy-1.6.6 decorator-5.1.1 executing-1.2.0 importlib-metadata-6.0.0 ipykernel-6.21.2 ipython-8.10.0 jedi-0.18.2 jupyter-client-8.0.2 jupyter-core-5.2.0 matplotlib-inline-0.1.6 nest-asyncio-1.5.6 parso-0.8.3 pexpect-4.8.0 pickleshare-0.7.5 platformdirs-3.0.0 prompt-toolkit-3.0.36 psutil-5.9.4 ptyprocess-0.7.0 pure-eval-0.2.2 pygments-2.14.0 pyzmq-25.0.0 stack-data-0.6.2 tornado-6.2 traitlets-5.9.0 wcwidth-0.2.6 zipp-3.13.0\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "pexpect",
                  "pickleshare",
                  "wcwidth"
                ]
              }
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n",
            "\n",
            "ResolvePackageNotFound: \n",
            "  - conda==22.9.0\n",
            "\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-bedgraphtobigwig\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
            "    libpng-1.6.39              |       h753d276_0         276 KB  conda-forge\n",
            "    mysql-connector-c-6.1.6    |                2         9.6 MB  bioconda\n",
            "    ucsc-bedgraphtobigwig-357  |                0         972 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        11.2 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2 \n",
            "  libpng             conda-forge/linux-64::libpng-1.6.39-h753d276_0 \n",
            "  mysql-connector-c  bioconda/linux-64::mysql-connector-c-6.1.6-2 \n",
            "  ucsc-bedgraphtobi~ bioconda/linux-64::ucsc-bedgraphtobigwig-357-0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "libpng-1.6.39        | 276 KB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "mysql-connector-c-6. | 9.6 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "libpng-1.6.39        | 276 KB    | :   6% 0.05797614287382475/1 [00:00<00:06,  6.74s/it]\n",
            "mysql-connector-c-6. | 9.6 MB    | :   0% 0.0016203364888226057/1 [00:00<04:09, 249.70s/it]\u001b[A\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | :   5% 0.05255779785906529/1 [00:00<00:07,  8.07s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "libpng-1.6.39        | 276 KB    | : 100% 1.0/1 [00:00<00:00,  6.74s/it]                \n",
            "mysql-connector-c-6. | 9.6 MB    | :  19% 0.18633869621459964/1 [00:00<00:01,  2.07s/it]   \u001b[A\n",
            "mysql-connector-c-6. | 9.6 MB    | :  70% 0.6967446901937204/1 [00:00<00:00,  1.74it/s] \u001b[A\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  1.87it/s]                \u001b[A\u001b[A\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  1.87it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "ucsc-bedgraphtobigwi | 972 KB    | : 100% 1.0/1 [00:00<00:00,  1.46it/s]                \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "ucsc-bedgraphtobigwi | 972 KB    | : 100% 1.0/1 [00:00<00:00,  1.46it/s]\u001b[A\u001b[A\u001b[A\n",
            "mysql-connector-c-6. | 9.6 MB    | : 100% 1.0/1 [00:02<00:00,  3.31s/it]               \u001b[A\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\bdone\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 21 not upgraded.\n",
            "Need to get 580 kB of archives.\n",
            "After this operation, 2,055 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 bedtools amd64 2.27.1+dfsg-4ubuntu1 [580 kB]\n",
            "Fetched 580 kB in 1s (773 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 128057 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.27.1+dfsg-4ubuntu1_amd64.deb ...\n",
            "Unpacking bedtools (2.27.1+dfsg-4ubuntu1) ...\n",
            "Setting up bedtools (2.27.1+dfsg-4ubuntu1) ...\n"
          ]
        }
      ],
      "source": [
        "! pip install pyGenomeTracks\n",
        "! pip install -U ipykernel\n",
        "! conda install -c bioconda ucsc-fetchchromsizes\n",
        "! conda install -c bioconda ucsc-bedgraphtobigwig\n",
        "! apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d s_8_cell.deduplicated.bismark.cov.gz\n",
        "!gzip -d s_epiblast.deduplicated.bismark.cov.gz\n",
        "!gzip -d s_ICM.deduplicated.bismark.cov.gz"
      ],
      "metadata": {
        "id": "oUsiUiIZxkoI"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!head s_8_cell.deduplicated.bismark.cov"
      ],
      "metadata": {
        "id": "3vkeb4zbz2uR",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6d36e5ed-b6b1-4f23-bb0f-8fa46175cec4"
      },
      "execution_count": 33,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr11\t3100031\t3100031\t50\t1\t1\n",
            "chr11\t3100048\t3100048\t66.6666666666667\t2\t1\n",
            "chr11\t3100049\t3100049\t50\t1\t1\n",
            "chr11\t3100224\t3100224\t87.5\t7\t1\n",
            "chr11\t3100225\t3100225\t53.3333333333333\t8\t7\n",
            "chr11\t3100286\t3100286\t66.6666666666667\t6\t3\n",
            "chr11\t3100287\t3100287\t33.3333333333333\t5\t10\n",
            "chr11\t3100310\t3100310\t62.5\t5\t3\n",
            "chr11\t3100311\t3100311\t50\t7\t7\n",
            "chr11\t3100349\t3100349\t55.5555555555556\t5\t4\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import matplotlib.pyplot as plt\n",
        "import seaborn as sns"
      ],
      "metadata": {
        "id": "wAx3ICEpzbGP"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "head = [\"chr\", \"start\", \"end\", \"M%\", \"met\", \"nonmet\"]"
      ],
      "metadata": {
        "id": "URneHG0Fzxcp"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "cell8 = pd.read_csv(\"s_8_cell.deduplicated.bismark.cov\", sep = \"\\t\", names = head)\n",
        "epiblast = pd.read_csv(\"s_epiblast.deduplicated.bismark.cov\", sep = \"\\t\", names = head)\n",
        "ICM = pd.read_csv(\"s_ICM.deduplicated.bismark.cov\", sep = \"\\t\", names = head)"
      ],
      "metadata": {
        "id": "u-qEWtvy0M-7"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "a = sns.histplot(data=cell8, kde=True, x=\"M%\", bins=10)\n",
        "a.set_title(\"8-cell\")\n",
        "a.set(ylim=(0,3500000))\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "id": "-pLk9pho033l",
        "outputId": "52ba8bcc-04a3-4626-ad27-6214bc5f0181"
      },
      "execution_count": 66,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZgcdb3v8fe3t9kzS2YSksxkIQkJISCQiGxHIqgHRAERFeSqeNVcjxsePV5FvZ6Dx3PU5/G4oiIK4ooLIEZEERBBlG0SIZCEQPZMttkzmX3p7/2jK3EYJplJMtU9mfq8nqef7q76TfW3Upn+TFX96lfm7oiISHTFcl2AiIjkloJARCTiFAQiIhGnIBARiTgFgYhIxCkIREQiTkEgkgNmtsXMXh28/g8z+0mua5LoUhCIDGFms83sHjNrMbPdZnaDmSVyXZdIWBQEIi/1baAemAacCpwHvD+nFYmESEEg8lJzgF+6e7e77wb+AJw0XEMzqzCzH5jZzmAP4q5B815vZk+ZWauZ/c3MTslS/SKHRUEg8lJfA640s0IzmwFcRCYMhvNjoJBMUEwBvgpgZqcBtwD/B5gMfBdYYWZ5IdcuctiOySAws1vMrN7Mnh1l+7eY2VozW2NmPwu7PjnmPUzmi70NqANqgbuGNjKzaWRC4n3u3uLufe7+UDB7OfBdd3/c3Qfc/YdAD3BmVtZA5DAck0EA3ApcOJqGZjYfuA44x91PAj4SYl1yjDOzGJm//u8EioBKoBz4kpndaGbtweNTQA3Q7O4twyxqFvCx4LBQq5m1Bu2nZ2dNREbvmOwJ4e4Pm9nswdPMbC7wLaAK6ATe6+7PAe8FvrX/l9Xd67NbrRxjKoCZwA3u3gP0mNkPgM+7+2LgffsbBnsEFWZW5u6tQ5azHfgvd/+vbBUucqSO1T2C4dwEfMjdlwD/RqbnB8AJwAlm9lcze8zMRrUnIdHk7o3AZuBfzCxhZmXAO4HVw7TdBfwe+LaZlZtZ0sxeGcz+HvA+M3uFZRSZ2cVmVpKtdREZrQkRBGZWDJwN/MrMniJzYm5aMDsBzAeWAVcB3wt+uUUO5nIyhx4bgA1AH/CvB2n79mD+c2S6nH4EwN1ryeyN3gC0BMu5JsyiRY6UHas3pgkODd3t7ovNbBKw3t2nDdPuRuBxd/9B8P4B4JPu/mQ26xURGa8mxB6Bu7cBm83szQDBrvjLgtl3kdkbwMwqyRwq2pSLOkVExqPQgsDM8s3sCTN7Oui2ef0wba4xs4bgopunzOw9o1z2bcCjwAIzqzOzdwNXA+82s6eBNcClQfN7gSYzWws8CHzc3ZvGYh1FRCaC0A4NmZkBRe7ebmZJ4BHgWnd/bFCba4Cl7v7BUIoQEZERhdZ91DMJ0x68TQaPY/OEhIjIBBbqdQRmFgdWAvPI9OV/fJhmbwq63D0P/Ku7bx9mOcvJXKlJUVHRkoULF4ZYtYjIxLNy5cpGd68abl5Weg0F3TV/Taaf/7ODpk8G2t29x8z+D/BWdz//UMtaunSp19bWhluwiMgEY2Yr3X3pcPOy0msouOryQYYMC+HuTcHVmwDfB5Zkox4REfmHMHsNVe2/cMvMCoDXkLnoZnCbwf3+LwHWhVWPiIgML8xzBNOAHwbnCWJkxne/28w+B9S6+wrgw2Z2CdAPNKMrL0VEsu6Yu7JY5whERA5fzs8RiIjI+KUgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIhGnIBARiTgFgYhIxCkIREQiTkEgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGICy0IzCzfzJ4ws6fNbI2ZXT9Mmzwz+4WZbTCzx81sdlj1iIjI8MLcI+gBznf3lwGnAhea2ZlD2rwbaHH3ecBXgS+FWI+IiAwjtCDwjPbgbTJ4+JBmlwI/DF7fDlxgZhZWTSIi8lKhniMws7iZPQXUA/e5++NDmswAtgO4ez+wF5g8zHKWm1mtmdU2NDSEWbKISOSEGgTuPuDupwLVwBlmtvgIl3OTuy9196VVVVVjW6SISMRlpdeQu7cCDwIXDpm1A6gBMLMEUAo0ZaMmERHJCLPXUJWZlQWvC4DXAM8NabYCeGfw+grgT+4+9DyCiIiEKBHisqcBPzSzOJnA+aW7321mnwNq3X0FcDPwYzPbADQDV4ZYj4iIDCO0IHD31cBpw0z/7KDX3cCbw6pBRERGpiuLRUQiTkEgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGIUxCIiEScgkBEJOIUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIhEXWhCYWY2ZPWhma81sjZldO0ybZWa218yeCh6fDaseEREZXiLEZfcDH3P3VWZWAqw0s/vcfe2Qdn9x99eHWIeIiBxCaHsE7r7L3VcFr/cB64AZYX2eiIgcmaycIzCz2cBpwOPDzD7LzJ42s9+b2UkH+fnlZlZrZrUNDQ0hVioiEj2hB4GZFQN3AB9x97Yhs1cBs9z9ZcA3gbuGW4a73+TuS919aVVVVbgFi4hETKhBYGZJMiHwU3e/c+h8d29z9/bg9T1A0swqw6xJREReLMxeQwbcDKxz968cpM1xQTvM7IygnqawahIRkZcKs9fQOcDbgWfM7Klg2qeAmQDufiNwBfAvZtYPdAFXuruHWJOIiAwRWhC4+yOAjdDmBuCGsGoQEZGR6cpiEZGIUxCIiEScgkBEJOIUBCIiEacgEBGJOAWBiEjEKQhERCJOQSAiEnEKAhGRiFMQiIhEnIJARCTiFAQiIhGnIBARiTgFgYhIxCkIREQiTkEgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMSFFgRmVmNmD5rZWjNbY2bXDtPGzOwbZrbBzFab2elh1SMiIsNLhLjsfuBj7r7KzEqAlWZ2n7uvHdTmImB+8HgF8J3gWUREsiS0PQJ33+Xuq4LX+4B1wIwhzS4FfuQZjwFlZjYtrJpEROSlsnKOwMxmA6cBjw+ZNQPYPuh9HS8NCxERCVHoQWBmxcAdwEfcve0Il7HczGrNrLahoWFsCxQRibhQg8DMkmRC4KfufucwTXYANYPeVwfTXsTdb3L3pe6+tKqqKpxiRUQiKsxeQwbcDKxz968cpNkK4B1B76Ezgb3uviusmkRE5KVG1WvIzM5x97+ONG2Ic4C3A8+Y2VPBtE8BMwHc/UbgHuB1wAagE3jX4ZUvIiJHa7TdR78JDO3jP9y0A9z9EcAOtVB3d+ADo6xBRERCcMggMLOzgLOBKjP76KBZk4B4mIWJiEh2jLRHkAKKg3Ylg6a3AVeEVZSIiGTPIYPA3R8CHjKzW919a5ZqEhGRLBrtOYI8M7sJmD34Z9z9/DCKEhGR7BltEPwKuBH4PjAQXjkiIpJtow2Cfnf/TqiViIhIToz2grLfmtn7zWyamVXsf4RamYiIZMVo9wjeGTx/fNA0B44f23JERCTbRhUE7j4n7EJERCQ3RjvExDuGm+7uPxrbckREJNtGe2jo5YNe5wMXAKsABYGIyDFutIeGPjT4vZmVAT8PpSIREcmqIx2GugPQeQMRkQlgtOcIfkumlxBkBps7EfhlWEWJiEj2jPYcwZcHve4Htrp7XQj1iIhIlo3q0FAw+NxzZEYgLQd6wyxKRESyZ1RBYGZvAZ4A3gy8BXjczDQMtYjIBDDaQ0OfBl7u7vUAZlYF3A/cHlZhIiKSHaPtNRTbHwKBpsP4WRERGcdGu0fwBzO7F7gteP9WMjeeFxGRY9xI9yyeB0x194+b2eXAucGsR4Gfhl2ciIiEb6Q9gq8B1wG4+53AnQBmdnIw7w2hViciIqEb6Tj/VHd/ZujEYNrsQ/2gmd1iZvVm9uxB5i8zs71m9lTw+OyoqxYRkTEz0h5B2SHmFYzws7cCN3Dogen+4u6vH2E5IiISopH2CGrN7L1DJ5rZe4CVh/pBd38YaD6K2kREJAtG2iP4CPBrM7uaf3zxLwVSwBvH4PPPMrOngZ3Av7n7mjFYpoiIHIZDBoG77wHONrNXAYuDyb9z9z+NwWevAma5e7uZvQ64C5g/XEMzWw4sB5g5c+YYfLSIiOxn7j5yqyNduNls4G53XzxCU8xsC7DU3RsP1W7p0qVeW1s7JvWJiESFma1096XDzcvZ1cFmdpyZWfD6jKCWplzVIyISVaO9sviwmdltwDKg0szqgH8HkgDufiNwBfAvZtYPdAFXepi7JyIiMqzQgsDdrxph/g1kupeKiEgOaeA4EZGIUxCIiEScgkBEJOIUBCIiEReZIEinnQfX14/cUEQkYiITBL+o3c67fvAkt6+sy3UpIiLjSmSC4M1Lqjln3mQ+deczrNyqsfBERPaLTBAk4jG+9bbTqSrJ44u/fy7X5YiIjBuRCQKAssIUl58+g5VbW9jb2ZfrckRExoVIBQHAsgVVpB0efqEh16WIiIwLkQuCU2vKKStM8uf1CgIREYhgEMRjxivnV/HQ8/Wk0xrjTkQkckEAmcNDje29rN3VlutSRERyLpJBsGRWOQDP7tib40pERHIvkkFQU15IcV6CNTu1RyAiEskgiMWMRdMm6dCQiAgRDQKARdMnsW5Xm04Yi0jkRTcIpk2is3eALU0duS5FRCSnohsE0ycB6PCQiEReZINg/tRiEjHTCWMRibzIBkFeIs68KcWsVRCISMRFNggATphawsaG9lyXISKSU5EOgjmVRexo7aK7byDXpYiI5ExoQWBmt5hZvZk9e5D5ZmbfMLMNZrbazE4Pq5aDOb6qCHfY1tyZ7Y8WERk3wtwjuBW48BDzLwLmB4/lwHdCrGVYx1cWA7CpQV1IRSS6QgsCd38YONQ9IS8FfuQZjwFlZjYtrHqGM7uyEIBNjTpPICLRlctzBDOA7YPe1wXTXsLMlptZrZnVNjSM3X0ESvKTVJXksVl7BCISYcfEyWJ3v8ndl7r70qqqqjFd9pzKIjY3KghEJLpyGQQ7gJpB76uDaVk1t0pBICLRlssgWAG8I+g9dCaw1913ZbuIOZVFNHX06mb2IhJZibAWbGa3AcuASjOrA/4dSAK4+43APcDrgA1AJ/CusGo5lDn7ew41tnPazPJclCAiklOhBYG7XzXCfAc+ENbnj9acyiIANjd2KAhEJJKOiZPFYZpZUUg8ZjpPICKRFfkgSCVi1JQX6KIyEYmsyAcBZA4PbdIegYhElIKAzAnjLY0dum2liESSgoDM4HNdfQPs2ded61JERLJOQQAcH/Qc0nkCEYkiBQEwpyoIAp0nEJEIilQQ1MychZm95DG9rIh0bzfXfvrzw84fi0fNzFm5Xn0RkWGFdkHZeFS3fRtf+eP6Yef97PFtHH/h27j0kx8P5bM/+toFoSxXRCa+gbRz08ObOHdeJSdXl4758iMVBIdSVpikfl9PrssQEXmRna1dfPBnq1i1rZXW845XEISpvDDFhoZ2BtJOPGa5LkdEBIDP/uZZntu9j69feSqXvGx6KJ8RqXMEh1JWmMQd2ro0CqmIjA9Pb2/l/nX1vH/ZXC49dQZm4fyRqiAIlBemAGjp7M1xJSIiGV+9/3nKCpNcc86cUD9HQRAoK0wC0Kr7EojIOLC1qYM/r2/gPefOoTgv3KP4CoJAfjJOQTKuPQIRGRd+90zmPl2XnTbsrdzHlIJgkLLCpPYIRGRc+N3qXZxaU0Z1eWHon6UgGKSsMElLl/YIRCS3tjR2sGZnG68/ZVpWPk9BMEh5YYqOngF6+9O5LkVEIuyeZzOHhS46WUGQdQdOGGuvQERy6KH1DSyaNokZZQVZ+TwFwSAHupB26DyBiORGZ28/f9/WyrnzK7P2mQqCQcoKkhjQrJ5DIpIjT25poXcgzdlzJ2ftMxUEgyTiMUoLkjR3KAhEJDf+tqGRZNw4Y05F1j4z1CAwswvNbL2ZbTCzTw4z/xozazCzp4LHe8KsZzQqilI0tysIRCQ3HtnQyGkzyylMZW8ouNCCwMziwLeAi4BFwFVmtmiYpr9w91ODx/fDqme0JhenaO3qZUD3LxaRLGvu6GXtrjbOnZe98wMQ7h7BGcAGd9/k7r3Az4FLQ/y8MVFRlCLt0KrzBCKSZY9ubMIdzpmXvfMDEG4QzAC2D3pfF0wb6k1mttrMbjezmuEWZGbLzazWzGobGhrCqPWAiqJMzyGdJxCRbPvrxkaK8xKcUl2W1c/N9cni3wKz3f0U4D7gh8M1cveb3H2puy+tqqoKtaD9XUgVBCKSbX/b0Mgr5lSQjGf3qznMT9sBDP4LvzqYdoC7N7n7/tuCfR9YEmI9o5IMeg41KQhEJIvqWjrZ0tTJOVk+PwDhBsGTwHwzm2NmKeBKYMXgBmY2+PrpS4B1IdYzahVFKe0RiMiwambOwszG/LHwvMsAeN8bX3XQNjUzZ4WyTqH1T3L3fjP7IHAvEAducfc1ZvY5oNbdVwAfNrNLgH6gGbgmrHoOR2Vxii1NHfQPpElkeRdNRMa3uu3b+Mof14/5cn//7C7qWrr44k/vPeidyD762gVj/rkQ8j2L3f0e4J4h0z476PV1wHVh1nAkppTk4w6NHb0cNyk/1+WIyATn7mxv7mJmRWFot6M8FP25O4wpJXkA1Ld157gSEYmCpo5euvoGqKnIziBzQ2Xv0rVjSEl+grxEjIZ9PSM3FgG2NXWyo7WLssIkJ06blOty5BizvbkTgJqK8G9CMxwFwTDMjCmT8qhXEMgIuvsG+Mp9z3PzI5sPXI1+8cnT+I9LTqIq2LMUGcm25k7KCpJMyk/m5PMVBAcxpSSfp7a1MpB24rHsH7OT8W8g7XzwZ6u4f109V51RwxtOmU7t1hZueHAD6/fs4873n52zX2w5dgyknR2tXSw8Lnd7kjpHcBBTSvIYcKepQ3sFMrwv/n4d96+r53OXnsQXLj+Fs+dV8uEL5nPru17OlsYOPvDTVRqzSka0e283fQOes/MDoCA4qKoDJ4wVBPJSf9vQyPf+spm3nzmLd5w1+0Xzzp5byecuXcxfXmjk1r9tyUl9cuzY3NRBzGBmjs4PgILgoMoKkhQk4+xo7cp1KTLOdPcN8KlfP8OsyYV8+uITh21z1Rk1LFtQxf/8cb3+D8khbWnsYHpZAXmJeM5qUBAchJlRXV5AXUsX7tq9l3+48aGNbGnq5L/feDL5yeF/ec2M/7x0Me5w/Yo1Wa5QjhVt3X00dfQyZ3JRTutQEBxCdXkB7T39tHX357oUGSf2tHXz3Yc2cfEp00YcE6amopAPnj+PP67dw2ObmrJUoRxLtjR2ADC7UkEwblWXZ47Z1bV05rgSGS/+54/rGUg7n7xw4ajav/vcOUwvzefzv1tLWieOZYhNjR2UFiQpL8xt7zIFwSGUF2bOE9S16BivwJqde/nVyjrecdasUV/4k5+M84mLFvLsjjZ+/fcdI/+AREZX7wDbmzuZN6U4J8NKDKYgOASdJ5D93J3P372OsoIkH7pg/kHbDTcy5WWn1dCz83muveUBYqn8UEauDGtUSgnPC/X7SDssmFqS61J0QdlIZlcW8UJ9O3vaejiuVAPQRdV9a/fw6KYmrr/kJEoLDr4bf7CRKXe0dnH7yjre8s0HecXxY38bwrBGpZTwPL+nnYrCFJXFqVyXoj2CkcytLCJmmfSWaOrtT/Pf96xjblURb3vFzCNaxoyyAuZNKaZ2awvtPep8EHX7uvvY0drFCVNzf1gIFAQjykvGmTW5iOf3tOvw0BgbSDu79nbR2N4zrq/A/dGjW9jS1MlnLl50VLcQPGfuZNLuPLpRPYiibnXdXgzGzQCFOjQ0CvOnFLO5sYPdbd1MK83dZeATQTrt/Pn5en786Fae2NxMR+8AAKUFSV55QhXXnD2LJbMqclzlPzTs6+EbD7zAP82vZNmCo7tfdllhilNryli1rZVTqkuZqntdRFLfQJpnduxlblUxkw5xmDGbFASjcHxVEYmY8cyOvQqCo7Chfh/X3fkMT25pYeqkPN60pJoTppYwkHae2bGX+9ft4bdP72TZgio+f9niA913c8Xdue7OZ+juT/Pvb1g0JrvwZ8yuYP2efdy3dg9XnlFDIjY+dsr7B9I8/EIDf9vQxPP17XT3DTClJI/FM0q5+ORpORseeSJau6uNnv40p80sy3UpBygIRiEvEWfx9FJW72jlzDmTx02KHyt6+gf49oMb+fafN1CUl+CLl5/Mm5ZUv+QwS2dvPz95bCtfu/8FXvvVh/m31y7gnWfPztnor7+qreP+dXv49OtOZN6UsenZkZeM8+qFU/nN0zt5bGMz587P/o3KB+vpH+Anj23jew9vYndbN6lEjAVTSyhIxXm6rpW7V+/ii79/jlctqOLj/7yQRdPHx6GMY1XfQJraLS0cNymfaeOo84mCYJROn1XG6h2trNzawqsWTsl1OceMlVtb+MQdq9lQ385lp07nM69fRGXx8OP0F6YSLH/lXC4+ZTqf/vUzfO7utfzm6Z184Y0nZ/0LaOXWZj5z17OceXwF//vcOWO67NmVRSyeMYmV21qoKsljwXHZ7z6YTjsrnt7Jl/+4nrqWLs6eO5nrLz2JZQuqXjTmzY7WLu5YWcfNj2zm4m/+hctOncFHX3OC9hCO0BObm2nv6eeixceNi5PE+ykIRqkkP3PnqTU721g8o1Q3HRlBR08/J771E9iCZQy0NdJ07w18/Uur+PphLKPwxPNYdcF7uehrTbQ9eRd7//ozvG/k0WCra2ayfdvWI679ud1tLP/RSqaV5fPtq5eEskdy3glVNHf0ct+6PRSk4kc/8qTFRv3Fkn/8Uspf+XZSU+fSs3sDrQ/dym1bnuK2Q/xMLK+ISa+4gjt6L+HOJ7fQtuq3tP3tF0yfMvmo/q2jpLG9h1XbWjjxuBKml42vQ8wKgsNw9tzJbG7s4A9rdnPly2uOqgfJeNPVO8DGhnY21LfTHNw/tbO3n4JknCkl+VSV5DGjvIBZkwsPOUpiw74e7lhVx/ce3kRs4fmcUl3KOXPnk7r8nCOqq7tvgEc2NLIm9iamv/ItLJlZzqLpkw5Zw9H0qf/z+no+9LO/U5SX4JZrXk5FUTh9vBOxGK8/eTp3rKrjN0/t4DUnTmXh0fQg8fSw1y8cmO3OtuZOntjczM693ZQWJDnz+AoWnD8Pu/qiUX9Me3c/j25qYm3icqacfQW777+Z1s5eygpz3xd+POvo6WfF0zspSMZHHKMqFxQEh6EwleC1i6Zy11M7ueeZXVy0eBqpxLETBu5Ow74etjZ3srmxI/PFv6ed5+v3BVdPv7h9zGBor86YwYzyAo6vLGZOZdGB8yXNHT2s2dnG6rq9DKSdf5pfyS8/+06uvXXFUdWcn4zz6hOncuK0STy6sYmHX2jkrxubmD25kJqKQiqL8igtTFKUih/VrvZzu9v4zp838pundrJgagk/eNfLQ/+rrSAV54ol1fx29U7uXbuHF+rbOWvu5IMeOjtc3X0D7GnrZltzJxsbOtjb1UdhKs6rFlRx0vTSI9rTKc5P8JpFUzm1poy/bmikZ9m7OPMLD7DshCmct6CKE6dNYuqkPFLxGO09/ezr7qe1s4+de7vY1drNrr1d7NrbTWdvP30DTt9AmqJUgvKiJBVFeUydlMeMsgKqywupLi/guNL8Y/4PrtbOXu5evYuu3gGuWFJNUd74+9oNtSIzuxD4OhAHvu/uXxwyPw/4EbAEaALe6u5bwqzpaM2aXMT5C6fw4HP1/Grlds47oSpnvVu6+wZo6eyluaOXlo4+mjt7aenopSV4bu7sY8UfHqAvliJWUEKsoJRY8h9fMt7fR19zHX1N2+lr3JZ5btrOQHsz6b5uGOjHEilihWXEiytIlh1HomI6+yqq2Vgxg0TZNGJ5hbinSXd30Ne4lZ7tz9Kx7mF+0rhtTNd1RlkBVyypZvfebtbuaguCrOPA/HjMKEjGSSViHPe/vszbb36ckvwEJXlJivMTFOclKMlPHAhuA9q6+9ne3MmqbS08v6edVCLGh8+fx/tfNe+gw0uPtfxknMtPq+bv21t4fFMzmxo7mFKSR3V5AeWFKQpScczAMNyd3oE0ff2Z597+9IHnyks/yV1/33HgfUdvP919aSAT3tXlhZx1/GTmTikak55KVSV5XHbaDD5x9YV84qYV/HHtbv6wZveIP1dZnMe00nyK8xIUpGIkY0ZHbz+bGztYubWFpo7eF/1BEjM4blI+M8oLmFFWQFlhikkFSSblJygtSAavkxTnJSjMi2eeU3EKU4mc32K2tz/TTfSJLc3EgDe8bPq47TIcWhCYWRz4FvAaoA540sxWuPvaQc3eDbS4+zwzuxL4EvDWsGoaKyfPKKU4L8GfnqvnjlU7mFycYlZFIZOL8yhKZb6MUvEYsZiBgwPJyTWs372PtDsDaaenf4CevjQ9/Wl6+gfo7ss89/Sn6elL09U3wL7uvgPDYLd39x94v/+vrK6+gYPWWFqQpKIoRW/aWHjyKeQnYxQk40zKT1JamKS0IElpfpJYbNFR/Vu4Ow7EzIDTgTcemBfGsAfHleZzXGk+7k5bdz+tnb20dvXR1tVHd1/mS3BHbyf7uvvZ2dpFe0/m327/9QpDlRcmOaW6jLcsreFNp1dTHtKhoEOJx4ylsyo4aVop63a1sbGhnae372VgFBcwpuKxzP+3qll09w+QiscoLEwyrTSfssIUVSWZv7LDuulJX+NWPv/GkwEjUTaVZOUs4kVlWDxJuqeTdG8n6e52Btoa6G9vYuvACFdVxxMkSqpIlE4hUTqF+KQptJVOZdOkKhKTqojlFxPLLx5VbQXJOEV5mVAoyktQlIpTGDwPfp+fiJNMGMlYjGTcSMQzz8l4jEQ8RipuJGIxkolMcOVVn0RdSyce/G7v/x0YSDtdfQO0dfVR39ZDXWsXA2lnZkUhFyycMq57G1pYV8ua2VnAf7j7PwfvrwNw9y8ManNv0OZRM0sAu4EqP0RRS5cu9dra2iOt6ZDHUQ9X/0CaNTvb2NDQzs7WrpccRjla6dXHFpIAAAZSSURBVN5u0r2deE8H6Z4u0r0dpHs68d5O0l3tDHS1ke5qY6Az85zuagum7QNPH1jOWK7z4fjoaxfk5LM/+s8nvmj9gczJ1FQBFouDGWB4bxfeP/a3Ih2LdR5IOx09/QfC3j1Tdiqe+UJKBV9W+w+H5ezfOgefm3antz/NZ6++AEsVEM8vxlL5xFKFWDIfSxUQSxVkpiWD51QBlizIDPqXCp6TQbv42P49HDMoL0xRU1HIgqklYzpG2Udfu+CIRzgws5XuvnTYeSEGwRXAhe7+nuD924FXuPsHB7V5NmhTF7zfGLRpHLKs5cDy4O0C4Ej/51UCjSO2mli0ztGgdY6Go1nnWe4+7OXx4++sxTDc/SbgpqNdjpnVHiwRJyqtczRonaMhrHUO83T8DqBm0PvqYNqwbYJDQ6VkThqLiEiWhBkETwLzzWyOmaWAK4GhfQlXAO8MXl8B/OlQ5wdERGTshXZoyN37zeyDwL1kuo/e4u5rzOxzQK27rwBuBn5sZhuAZjJhEaajPrx0DNI6R4PWORpCWefQThaLiMix4di+ZE9ERI6agkBEJOIiEwRmdqGZrTezDWb2yVzXEwYzqzGzB81srZmtMbNrg+kVZnafmb0QPJfnutaxZGZxM/u7md0dvJ9jZo8H2/oXQWeFCcPMyszsdjN7zszWmdlZEdjG/xr8n37WzG4zs/yJtp3N7BYzqw+ur9o/bdjtahnfCNZ9tZmdfjSfHYkgGDTcxUXAIuAqMzu6sRXGp37gY+6+CDgT+ECwnp8EHnD3+cADwfuJ5Fpg3aD3XwK+6u7zgBYyQ5lMJF8H/uDuC4GXkVn3CbuNzWwG8GFgqbsvJtP5ZP+QNBNpO98KXDhk2sG260XA/OCxHPjO0XxwJIIAOAPY4O6b3L0X+DlwaY5rGnPuvsvdVwWv95H5gphBZl1/GDT7IXBZbioce2ZWDVwMfD94b8D5wO1Bk4m2vqXAK8n0uMPde929lQm8jQMJoCC43qgQ2MUE287u/jCZ3pODHWy7Xgr8yDMeA8rMbNqRfnZUgmAGsH3Q+7pg2oRlZrOB04DHganuviuYtRuYmqOywvA14P8C+wcXmgy0uvv+0c0m2raeAzQAPwgOh33fzIqYwNvY3XcAXwa2kQmAvcBKJvZ23u9g23VMv9OiEgSRYmbFwB3AR9y9bfC84IK9CdFn2MxeD9S7+8pc15JFCTLDvH7H3U8DOhhyGGgibWOA4Lj4pWRCcDpQxEsPoUx4YW7XqATBaIa7mBDMLEkmBH7q7ncGk/fs320MnutzVd8YOwe4xMy2kDncdz6Z4+dlwSEEmHjbug6oc/fHg/e3kwmGibqNAV4NbHb3BnfvA+4ks+0n8nbe72DbdUy/06ISBKMZ7uKYFxwfvxlY5+5fGTRr8FAe7wR+k+3awuDu17l7tbvPJrNN/+TuVwMPkhmyBCbQ+gK4+25gu5ntv9nDBcBaJug2DmwDzjSzwuD/+P51nrDbeZCDbdcVwDuC3kNnAnsHHUI6fO4eiQfwOuB5YCPw6VzXE9I6nktm13E18FTweB2Z4+YPAC8A9wMVua41hHVfBtwdvD4eeALYAPwKyMt1fWO8rqcCtcF2vgson+jbGLgeeA54FvgxkDfRtjNwG5lzIH1k9vzefbDtSuYme98Kvs+eIdOj6og/W0NMiIhEXFQODYmIyEEoCEREIk5BICIScQoCEZGIUxCIiEScgkDkMJiZm9lPBr1PmFnDoJFP3xSMkvkXM5scTJtrZr/IVc0iI1EQiByeDmCxmRUE71/Di6/o/BDwcuC7wNuCaZ8HPpO1CkUOk4JA5PDdQ2bEU4CryFwItF+azMVOhUCfmf0TsNvdX8huiSKjpyAQOXw/B640s3zgFDIjvO73BTJXgL6BTED8P+A/s16hyGFQEIgcJndfDcwmszdwz5B597n7End/A5kRM+8BTgjuKPY9MyvMesEiI1AQiByZFWTGyL9tuJnBF/41ZMaDuZ7MgGGPAFdnqT6RUUuM3EREhnELmRujPGNmy4aZ/3HgG+7eF5xYdjLnD7RHIOOOgkDkCLh7HfCN4eaZ2XTgDHe/Ppj0TTJDobdyjN9OUSYmjT4qIhJxOkcgIhJxCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMT9f34YZGMQUk8BAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "a = sns.histplot(data=epiblast, kde=True, x=\"M%\", bins=10)\n",
        "a.set_title(\"epiblast\")\n",
        "a.set(ylim=(0,2000000))\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "o41PPsDZ2IWd",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "outputId": "ebdbfed5-8432-4ce4-cd12-f837806d2602"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxddZ3/8dfn3ps9adKk6Z7uG8hOgLI5COoAI+KCgjIs80NwXMYNnZ/O4rjMb8Zx5oEzjI6IgKIiguhg9ccPBUVBgdKWodDSFtpS2pS2SZum2ZO7fH5/3BMMIW3SNufem5z38/G4j9xzzjfnfE5Pej/3u5zvMXdHRESiK5bvAEREJL+UCEREIk6JQEQk4pQIREQiTolARCTilAhERCJOiUBklMxsvZmdF7z/gpn94BBlt5nZm3MWnMhRSOQ7AJHxwt3fkMvjmdl3gSZ3/7tcHleiRzUCEZGIUyKQyDGzmWb2EzNrMbOXzOxjwfovmNl9ZnaPmXWY2dNmduKg3xva3FN6sLJDjne6mT1hZm1mtsvMvm5mxcE2M7OvmVmzmbWb2XNmdpyZ3QBcCfy1mXWa2c9D/CeRiFMikEgxsxjwc2AtMAu4APiEmf1pUORS4MdALfBD4H4zKzrI7kZbNg18EpgCnBkc88PBtrcCbwSWANXAe4F97n4rcBfwVXevdPdLjvikRUYwLhOBmd0RfINaN8ry7zWz54POvh+GHZ8UtNOAenf/krv3u/tW4NvAFcH2Ne5+n7sngZuAUmD5QfY1qrLuvsbdn3T3lLtvA74F/EmwOQlUAcsAc/cN7r5rbE5VZHTGZSIAvgtcOJqCZrYY+BxwdtDZ94kQ45LCNxeYGTTTtJlZG/A3wLRg+46Bgu6eAZqAmQfZ16jKmtkSM/uFme02s3bgn8jWDnD33wBfB74BNJvZrWY26WhPUuRwjMtE4O6PAq2D15nZQjN70MzWmNljZrYs2HQ98A133x/8bnOOw5XCsgN4yd1rBr2q3P3iYHvDQMGgGWk28MpB9jXast8ENgKL3X0S2cRjAxvd/WZ3PxU4lmwT0WcGNh3JCYocrnGZCA7iVuCvgv9Qnwb+K1i/BFhiZn8wsyfNbFQ1CZmwngI6zOx/m1mZmcWDztnTgu2nmtm7zCxBtvbYBzx5kH2NtmwV0A50Bl9QPjSwwcxOM7Mzgr6FLqAXyASb9wALju50RUY2IRKBmVUCZwE/NrNnyLbBzgg2J4DFwHnA+4Bvm1lNPuKU/HP3NPA24CTgJWAvcBvZjlqAnwGXA/uBq4B3BX0Awxlt2U8D7wc6yPZH3DNo26Rg3X7gZWAf8K/BttuBY4MmrPsP+2RFRsnG64NpzGwe8At3Py5oU93k7jOGKXcLsNLdvxMs/xr4rLuvymW8UvjM7AvAInf/83zHIpJLE6JG4O7twEtm9h54dWz2wJju+8nWBjCzKWSbirbmI04RkUIUWiIws1Ize8rM1gbDNr84TJlrg5t6ngleHxjlvu8GngCWmlmTmV1H9uab68xsLbCe7BhvgF8C+8zseeAR4DPuvm8szlFEZCIIrWnIzAyocPfOoCPs98DH3f3JQWWuBRrd/aOhBCEiIiMKbdI5z2aYzmCxKHiNzw4JEZEJLNTZR80sDqwBFpEdy79ymGLvNrM3Ai8An3T3HUMLBPOu3ABQUVFx6rJly4YWERGRQ1izZs1ed68fbltORg0FwzX/m+w4/3WD1tcBne7eZ2YfBC539/MPta/GxkZfvXp1uAGLiEwwZrbG3RuH25aTUUPu3ka2o/bCIev3uXtfsHgbcGou4hERkT8Kc9RQ/cCNW2ZWBryF7G32g8sMHvf/dmBDWPGIiMjwwuwjmAHcGfQTxIB73f0XZvYlYLW7rwA+ZmZvB1Jk5w66NsR4RERkGOPuzmL1EYiIHL689xGIiEjhUiIQEYk4JQIRkYhTIhARiTglAhGRiFMiEBGJOCUCEZGIUyIQEYk4JQIRkYhTIhARiTglAhGRiFMiEBGJOCUCEZGIUyIQEYk4JQIRkYhTIhARiTglAhGRiFMiEBGJOCUCEZGIUyIQEYk4JQIRkYhTIhARibjQEoGZlZrZU2a21szWm9kXhylTYmb3mNlmM1tpZvPCikdERIYXZo2gDzjf3U8ETgIuNLPlQ8pcB+x390XA14B/CTEeEREZRmiJwLM6g8Wi4OVDil0K3Bm8vw+4wMwsrJhEROT1Qu0jMLO4mT0DNAMPufvKIUVmATsA3D0FHADqhtnPDWa22sxWt7S0hBmyiEjkhJoI3D3t7icBs4HTzey4I9zPre7e6O6N9fX1YxukiEjE5WTUkLu3AY8AFw7ZtBNoADCzBFAN7MtFTCIikhXmqKF6M6sJ3pcBbwE2Dim2ArgmeH8Z8Bt3H9qPICIiIUqEuO8ZwJ1mFiebcO5191+Y2ZeA1e6+Argd+L6ZbQZagStCjEdERIYRWiJw92eBk4dZ//lB73uB94QVg4iIjEx3FouIRJwSgYhIxCkRiIhEnBKBiEjEKRGIiEScEoGISMQpEYiIRJwSgYhIxCkRiIhEnBKBiEjEKRGIiEScEoGISMQpEYiIRJwSgYhIxCkRiIhEnBKBiEjEKRGIiEScEoGISMQpEYiIRJwSgYhIxCkRiIhEnBKBiEjEhZYIzKzBzB4xs+fNbL2ZfXyYMueZ2QEzeyZ4fT6seEREZHiJEPedAm5096fNrApYY2YPufvzQ8o95u5vCzEOERE5hNBqBO6+y92fDt53ABuAWWEdT0REjkxO+gjMbB5wMrBymM1nmtlaM/t/ZvaGg/z+DWa22sxWt7S0hBipiEj0hJ4IzKwS+AnwCXdvH7L5aWCuu58I/Cdw/3D7cPdb3b3R3Rvr6+vDDVhEJGJCTQRmVkQ2Cdzl7j8dut3d2929M3j/AFBkZlPCjElERF4rzFFDBtwObHD3mw5SZnpQDjM7PYhnX1gxiYjI64U5auhs4CrgOTN7Jlj3N8AcAHe/BbgM+JCZpYAe4Ap39xBjEhGRIUJLBO7+e8BGKPN14OthxSAiIiPTncUiIhGnRCAiEnFKBCIiEadEICIScUoEIiIRp0QgIhJxSgQiIuPAVbev5Icrt4eybyUCEZEC99LeLh57cS/JdCaU/SsRiIgUuN9uagbgvKXhTLqpRCAiUuB+u6mFBVMqmFtXEcr+lQhERApYT3+aJ7bu409Cqg2AEoGISEF7cus++lMZzls6NbRjKBGIiBSwNS/vJx4zzphfG9oxlAhERArY7vZeplaVUFoUD+0YSgQiIgVsT3svUyeVhnoMJQIRkQLW3N7HtKqSUI+hRCAiUsB2t/cyvVo1AhGRSOpNpjnQk2SamoZERKKpub0PgKlqGhIRiaY9Hb0AqhGIiETV7gPZRKA+AhGRiNrTHtQIqsZpIjCzBjN7xMyeN7P1ZvbxYcqYmd1sZpvN7FkzOyWseERExpvmjj5KEjEmlSVCPU6Ye08BN7r702ZWBawxs4fc/flBZS4CFgevM4BvBj9FRCJvTzB01MxCPU5oNQJ33+XuTwfvO4ANwKwhxS4FvudZTwI1ZjYjrJhERMaTPe29oTcLQY76CMxsHnAysHLIplnAjkHLTbw+WYiIRNKe9j6mTgp36CjkIBGYWSXwE+AT7t5+hPu4wcxWm9nqlpaWsQ1QRKQAuXu2RhDy0FEIORGYWRHZJHCXu/90mCI7gYZBy7ODda/h7re6e6O7N9bXh/dwBhGRQtGTTNPdn6Y+5JvJINxRQwbcDmxw95sOUmwFcHUwemg5cMDdd4UVk4jIeNHWnQSgpqwo9GONatSQmZ3t7n8Yad0QZwNXAc+Z2TPBur8B5gC4+y3AA8DFwGagG/iLwwtfRGRiejURlBdIIgD+Exg6xn+4da9y998Dhxzz5O4OfGSUMYiIRMaBnmwimJTvGoGZnQmcBdSb2acGbZoEhPe4HBGRiBtIBNX5TgRAMVAZlKsatL4duCysoEREou5ATz8ANeXFoR/rkInA3X8H/M7MvuvuL4cejYiIAIVVIxhQYma3AvMG/467nx9GUCIiUdfWnSQRMyqKw2+FH20i+DFwC3AbkA4vHBERgWyNoLqsKPR5hmD0iSDl7t8MNRIREXnVQCLIhdHeUPZzM/uwmc0ws9qBV6iRiYhE2IGeJNU5uIcARl8juCb4+ZlB6xxYMLbhiIgIZBNBbUX4I4ZglInA3eeHHYiIiPzRgZ4k86dU5ORYo51i4urh1rv798Y2HBERgeyooVzMMwSjbxo6bdD7UuAC4GlAiUBEZIxlMk57b+46i0fbNPRXg5fNrAb4USgRiYhEXEdvCneozsFdxXDk01B3Aeo3EBEJQS7vKobR9xH8nOwoIchONncMcG9YQYmIRFlBJgLg3wa9TwEvu3tTCPGIiERe26sTzhXQDWXB5HMbyc5AOhnoDzMoEZEoy3WNYFSJwMzeCzwFvAd4L7DSzDQNtYhICHL5mEoYfdPQ3wKnuXszgJnVAw8D94UVmIhIVOXy6WQw+lFDsYEkENh3GL8rIiKHob0nSUkiRmlRbh4EOdoawYNm9kvg7mD5crIPnhcRkTHW1p3MWUcxjPzM4kXANHf/jJm9Czgn2PQEcFfYwYmIRFEup6CGkWsE/w58DsDdfwr8FMDMjg+2XRJqdCIiEdTW05/TRDBSO/80d39u6Mpg3bxD/aKZ3WFmzWa27iDbzzOzA2b2TPD6/KijFhGZwA70pKguy830EjByIqg5xLayEX73u8CFI5R5zN1PCl5fGqGsiEgktOe4aWikRLDazK4futLMPgCsOdQvuvujQOtRxCYiEklt3f2F01kMfAL4bzO7kj9+8DcCxcA7x+D4Z5rZWuAV4NPuvn4M9ikiMm4l0xm6+tOF01ns7nuAs8zsTcBxwer/6+6/GYNjPw3MdfdOM7sYuB9YPFxBM7sBuAFgzpw5Y3BoEZHC1J7j6SVg9M8jeAR4ZCwP7O7tg94/YGb/ZWZT3H3vMGVvBW4FaGxs9KHbRUQmirYgEeSyaShvdweb2XQzs+D96UEs+/IVj4hIIcj19BIw+juLD5uZ3Q2cB0wxsybgH4AiAHe/BbgM+JCZpYAe4Ap317d9EYm0A90F2jR0JNz9fSNs/zrw9bCOLyIyHg3UCHI18yho4jgRkYKS62cRgBKBiEhBactD05ASgYhIATnQk6SyJEEinruPZyUCEZECkusJ50CJQESkoOR6niFQIhARKSi5fhYBKBGIiBSUtm4lAhGRSDvQk9vHVIISgYhIQVHTkIhIhPUm0/SlMlSrRiAiEk35uKsYlAhERApGPu4qBiUCEZGC8ccJ53L34HpQIhARKRhqGhIRibi27n5AiUBEJLJerRFo1JCISDQd6EliBlUloT0zbFhKBCIiBWLgZrJYzHJ6XCUCEZECkY+7ikGJQESkYORjwjlQIhAROSwNc+ZiZqG8fvXbx1j52CMH3d4wZ24o55TbHgkRkXGuacd2bvrVplD2fefj25haVcJFHxh+/59669JQjqsagYhIgehLZSgpiuf8uKElAjO7w8yazWzdQbabmd1sZpvN7FkzOyWsWERECp2705tKU1qU++/nYR7xu8CFh9h+EbA4eN0AfDPEWEREClp/OoM7lCQmUI3A3R8FWg9R5FLge571JFBjZjPCikdEpJD1JTMAE65GMJJZwI5By03ButcxsxvMbLWZrW5paclJcCIiudSXyiaCCVUjGEvufqu7N7p7Y319fb7DEREZc73JNBC9GsFOoGHQ8uxgnYhI5PSmsokgajWCFcDVweih5cABd9+Vx3hERPImn30Eod1QZmZ3A+cBU8ysCfgHoAjA3W8BHgAuBjYD3cBfhBWLiEihy2eNILRE4O7vG2G7Ax8J6/giIuNJXzJDzKAontuZR2GcdBaLiEx0vck0JYk4ZkoEIiKR1JNMU1ac+2YhUCIQESkIPck0ZXmYZwiUCERECkJPvxKBiEikqWlIRCTCMu70JjOqEYiIRNXA9BKqEYiIRFRPf5AIVCMQEYmmHtUIRESiTTUCEZGIG6gRlKtGICISTQM1glLVCEREoqknmaYkESMey/08Q6BEICKSd/m8qxiUCERE8q47j3cVgxKBiEje5XPCOVAiEBHJu55+1Qhyoj+V4Sdrmsg+GE1EpDC4O72qEeTGT59u4sYfr+Xvf7aOdEbJQEQKQ18qQ8bzd1cxhPjM4kJz+WkNvLSvi2/9bisliTh//7Zj8x2SiMir9xCUq0YQPjPjcxcdw7tOnsUPV26nozeZ75BEROjsSwFQWZq/7+WRSQQDrjpzLj3JNL94dle+QxER+WMiKJmgicDMLjSzTWa22cw+O8z2a82sxcyeCV4fCDMegJMaalg6rYofrdoR9qFEREbUMZETgZnFgW8AFwHHAu8zs+Ea5u9x95OC121hxTMoLi4/rYG1O9p4YU9H2IcTETmkzt4UpUUxEvH8NdCEeeTTgc3uvtXd+4EfAZeGeLxRu/j4GQD8dlNzniMRkajr7EvltTYA4SaCWcDg9pemYN1Q7zazZ83sPjNrGG5HZnaDma02s9UtLS1HHdj06lKWTKvksRf3HvW+RESORmdviqrSorzGkO/O4p8D89z9BOAh4M7hCrn7re7e6O6N9fX1Y3LgcxfX89RLra8+K1REJB8meo1gJzD4G/7sYN2r3H2fu/cFi7cBp4YYz2ucu3gKfakMq7a15uqQIiKvkUpn6EmmJ3QiWAUsNrP5ZlYMXAGsGFzAzGYMWnw7sCHEeF7jjPl1FMdjah4SkbwphHsIIMQ7i909ZWYfBX4JxIE73H29mX0JWO3uK4CPmdnbgRTQClwbVjxDlRXHaZw3WYlARPKmEO4hgJCnmHD3B4AHhqz7/KD3nwM+F2YMh3LmgjpuevgF2rr7qSkvzlcYIhJRnb3ZRFA1gZuGCt7yhXW4w5Nb1U8gIrnXUSBNQ5FOBCfOrqGsKM6TW/flOxQRiaDO3hQliRhFebyZDCKeCIoTMRrnTVYiEJG8aO9NUpXn2gBEPBEALF9Qx8bdHbR29ec7FBGJmNaufmoLoH9SiWBBHQArVSsQmfAyBfRQqmQ6Q3tvitqK/CeC/NdJ8uyE2dWUF8d5Yus+Ljp+xsi/ICIFoWHOPHa1dRKvrMMSxVgsDrE4ZjGsuIzEpCnEq+pJTKonPin4WV6Np5Kku9tIte2mf/dmeneso/fltXiyN6fx7w9aIZQICkBRPEbjvFr1E4iME80dvdz6u634JV9kdtWUQ5YtjseoKktQVZKgqrSI8uI4qYzT3VfL/u4GWuadQPr0dxKPGQvrKzhhVg2zJpcdcp+feuvSMTmPgebousqSMdnf0Yh8IgBYvqCWrz64ib2dfUwpgIsiMpF19aX40aod/HZTMzEzTp07mfefMWfE/3uZjPOdx7dx06820Z/O0L/rRd7ceAxVpQmKYjFiZsRiYBiJuFFVmqAkcejHP6YzzittPWxu6WTT7g5e2NPJzOpSzlk8hRnVh04IR2tfVz8xg+qy/E44BxFLBA1z5tK0Y/vr1hfPWMKMq29ibuMFdG/6QyjHnt0whx3bXw5l3yJjpasvxcMb9rC1pYtjZkzircdOIxazMdv/c00HuO7OVTR39LF0WhXxmHHTQy/wX7/dzMcuWMz15y4Ydihlc0cvN967lsde3Mubltbz+UvewIL6P+PED119VPHEY0ZDbTkNteWcu2gK619pZ9W2Vu5d3cSxMybxJ0vqKU6E05Xa2pW9kTU+hv++RypSiaBpx3Zu+tWm161PZ5xvPbqF8z/yFc5fNjWUY49VdVIkLC/s6eBDP1jDlpauV9ctnlrJt69uZN6UiqPe/x827+X6761mcnkx9/3lmTTOqwVgc3Mn//rLjXz1wU2seOYV/s87j+fUuZOB7P/NB57bxRdWrKerP8U/vuM4rjxjDmZj/+GZiMc4saGGY2ZM4qltrTz98n52tvVw0XHTmTapdMyP19rVz5SqwmiBiFQiOJh4zJhZU8b21m7cPZQ/MpnY+lMZOnqTlBbFqcjzdAFHYtveLt79zccpScT4zl+cxpkL6njo+T18/mfreO+3nuCH15/BoqlVR7z/dTsPcMP3VtMwuZzvX3c6Uwd9sC6aWsm3rmrkV+t38w8r1vPubz7OsulVzKktZ9OeDl7e180xMyZx8xUnsXjakccwWsWJGOcsmsL8ugoeXL+be1fv4OxFUzi5oWbMPhtS6QwHepIsmR7++YzG+PuLDcn8KRX8dlML+7uTBdGLL+ODu3Pfmib+6YEN7O9OUhQ3rjtnAR+7YBHlxePjv1dvMs2H7nqamBn//eGzaagtB+CSE2eydHoVV962kqtvf4r7P3o2U6sO/5vxjtZu/uK7q6guK+LO//XaJDDYW98wnbMWTeGeVTv45frdbNvXxdy6Cj7zp0u56LgZOW9CmTW5jCvPmMPDG/bw2It7ae3q501Lx6bFoLWrHwfqCuSzZnz8pebAgiARbGnppLaiNt/hyDjg7vz9z9bxgye30zh3MpecOJO1O9q45XdbeGLrPn74gTPGRe3giz9fz4Zd7dxxbeOrSWDAkmlVfOfa07jslsf54PfXcPf1yyktOnQH7GBt3f1c+52n6EumuetDZzG9+tCJpLIkwXXnzOe6c+Yf0bmMtdKiOH92/Aye2LqPVdv2096bxIrLR/7FEezY3wPAjBH+PXIl8jeUDagqLWJqVQlbB7WPihzKzb/ezA+e3M4Nb1zAvR88k2vOmsdNl5/ELX9+Kut2HuCD319DfyqT7zAP6adPN3H3Uzv48HkLOX/ZtGHLHDermq+99yT+Z3sbn/3Js7iP7qas9t4k13xnFTtae/j21Y0syUGzThjMjLMWTuHNx0xl5/4epl/1r7T3Jo9qn9tbu6mtKM77IyoHKBEMsrC+kt3tvXQFMwKKHMwjG5v52sMv8K5TZvG5i5a9ZmTNhcdN5yvvOp7fb97Lv/5yYx6jPLQX9nTwt/+9jjPm1/Kptyw5ZNmLjp/Bp9+6hPufeYWvPfziq+sb5szFzF73ipdNYsn1/8Ez2/bSdO8XWL5wyrDljuaVa2+YWc07TppForKOe1ftoLnjyG5AS6Uz7GzrYW7t0dcsxkrh11tzaGF9BU9s3cemPR2cMmdyvsORArWnvZcbf7yWZdOr+Kd3Hj/sh9J7Ght4tukA337sJZYvqOOCY4b/tj1WDjY0+mCsqJQZ13yNWEklP/na+7n3L/eP6vfqLv44NwNf/vI/0vbY9wFeNxJvT3svDzy3i67+NBe+YTqL3vr9Ucd1OPIxEq+htpzdd/01kz58K/etaeLi42cwr+7wRlTtbOshnXHm1CkRFKS6yhJmVJfyXNOBMR0hIBNHOuN88p5n6OlP8/X3n3zI9vK//bNjWPPyfm788Voe+Ni5zKwJ7walgw2NHs7AkMyX9nbxzpNn0XDpk6M+TsadRzY2s+6syznm4mt4+hsfJ51xYga723tZt7Od53e1U1ES57JTZo/YJzAeJfdu5/LTGljxzCusWPsK5y+bynEzq0f9+9tbu4mbMSvEv4fDpaahIU6YXU1bT5Ltrd35DkUK0C2/28LjW/bxhbcfO+JwytKiON+48hSSqQx/dff/kEyPfX9Bd3+KLS2dJGpn05NMj1g+lcnw0PN72Lq3iz9ZWv+6zuGRxMw4f9lULlg2ldbOfqZf+S/81283c/NvNnPv6iY27e7g5IYarjpj7oRMAgMqSxJcdups5tSW8+sNzfz+xb2jmtAumc6wcXcHs2vL8v4MgsFUIxhi0dRKHn1hL2ubDjD3MKt8MrE99mILNz30Am87YQbvbWwYtsxwTTTly86l69L/zfS3fYLWB//zKKMwShqOo2LZ2ZTOO5mi2lkAzLr+Fm59dCtlRXGmTSphZk0ZM6vLmDaphEQ8RsazUyn87oUW9nb2c9bCOk6cXXNkEZhx3KxqFk+t5Mufup4LP/5V4mZMLi9ifn3FiNM6TBTFiRiXnDCTR19oYc32/exu7+VP3zDtkB3AzzYdoLs/zWlzC2tkohLBEIlYjBMbqnlyayvbW7uZU0AdOhOFu/NicyfPNh1g5/4eEnFj9uQyGufVFlR1ebAtLZ185K6nWVRfyVfefcJBmw0P1kTz+Ja9rOJPefM73scZ82sPu9mxtaufDbva2bSng47eFEVxY/bkcqZPKmVSaYIffOVG3nXjV9nX2c/uA71s25edRDFmUFGSoC+VoT+VoawoziUnzGBBfeXh/yMMUVIUp3vTHzhn0aEnfpvI4jHjTcumMr26lEc2NfODJ7dz9qI6jptVTWzINe5Npln9citza8tHnNgu15QIhnHqnMls2NXBIxubufKMOSQKqAo3nu3t7OOeVTv4yZomtu4dfpjuafMm897GBi4+fkbBjMHfsKudq+94iqJ4jNuuaaTyCOI6c0Ednb0pVr7USnd/mjcumUIidui/q/3d/Wxu7mRzcyfNHX0YMKeunLMXTmFBfcVrmha6Nzz6mgEOPck0u9p6eOVAdhRcUTzG7MllzKurCG3unCg7ZsYkZlSX8vCGZh7Z1MKzTQc4qaGGRVMrKUnEaO3q54F1u+lPZThzYV2+w32dwvifVmAS8RhvWlrP/c+8wsMbm7MTb0Wg47inP82O/d28vK+b3e29r36DnFpVwtRJJcyeXH7Yd11nMs7jW/Zx96rt/Gr9bpJpZ/mCWq47dz7LF9Qxp7acdMbZ3NzJoy+2cN/qJj5z37N8YcV63nbCTN55yiwa504+ZDJu6+5ne2s27paOPjLuuGe/rU2pKmFqVQnTJpUyo7r0sG6GymScu1dt5ysPbKSiJMEPbzjjsNvUB5gZbzl2GuUlCda8vJ+X93Vx6tzJzKktp7IkgQMdvSl2t/eyq62HnW097O/OjlWfPqmUcxdPYem0qlEnx7KiOAvqK8fkm7+MTk15Me8+ZRabmztZ+VIrv97YzK83NlMUN5JppyQR4x0nzQpl3qKjFWoiMLMLgf8A4sBt7v6VIdtLgO8BpwL7gMvdfVuYMY3W3LoKzlpYx+Nb9pFKZ3jT0qk5+Ybq7nT3p+nqT9HdF/zsT9PZN2i5L0VfKkMq46TSTioz8D5DMu2kM9l1A+8HOilLEsi+QkMAAAjFSURBVHFKimIUx2MUJ2J09CZp6ehnb2cfO9t6aOnoGzG+mvIiFtZXsrC+ggX1lSysr2RGdSkliRilRdm53lu7+tjS0sWn//kbZKYuJVE9lXRPO13rfkPH2ge5Z18T9xziGCWzjqHyhLdyd+c53LN6B+neTpJ7tpBs240n+4iVVBCvqMm+qqcRLx39h11tRTEza0qZWV2WbUevKWVGdRlTKksoK46TSmdo6ehj3SsH+MWzu3h5XzfLF9Tyb+85kdmTj66Z0Mw4Z9EUGiaX8YfN+3hkU8vw55+IMb26lBNm17CwvqJgbjqSkZkZi6dVsWhq9p6kV9p66ehNMrmimPlTKphUoNcytE82M4sD3wDeAjQBq8xshbs/P6jYdcB+d19kZlcA/wJcHlZMh+u0ebXEzfjDlr1sb93GwvpKZk8uo7qsiOJ4jFjMiJuRcc9+EAcfxungfTL4YE6mM9Sce1V2BsW+1Gs+6Dv7UnT3p+jqT9Pdl6I7mWaUN26+KmbZvo1E3IjHjKJ4jETMSMSMnU07SPVnP+AtUYTFi7FEEcSL8L5u0t1t2Vd7C8n9u0gd2EOqbTfp9hY81Y8VlxOvnEy8YjKJmul01M1md+1snqqdTbxyhHst5pzKkln1LJlWxcL6hSTedirwmVGfV38qw8utXWzf183eqVPo6E1mv1kVxSgvjlNenKCqNEF1WdGrr4qSBH/3rtPB01gskU0WlbXEK+uIT5pCR1U9OyfVE580hcSkemIlww8IcM/Q+/JzdK59kHs2PsY9Hxx12COaW1fBnNpyWrv6eaWtl95UdrRPZUmCqVUl1FYUa+jyOGdmzKguC/2ZBmMlzK+4pwOb3X0rgJn9CLgUGJwILgW+ELy/D/i6mZmP9h72HDhl7mQW1Ffw1LZWtu3tZuPujiPaz6Tll3H7b9bh/T14spdMXw+e7CHT3xv87MH7e8kkg5/9PXh/D5lkb/bnwO8F5Tzdj2fSkE4Dh/7nGu348sPVm0yzv7uf7v70q8nPyA6brCkv4svvPJFP/nLDEe+/OBFj8dQqFh/mrJfe1znqc+5LpenoTdGbTJNMZ8fDlxXFqSkvpjixFLjssI492puczIy6ypKCeDqViIX1mWtmlwEXuvsHguWrgDPc/aODyqwLyjQFy1uCMnuH7OsG4IZgcSlwpJ9sU4C9I5aaWHTO0aBzjoajOee57l4/3IZx0Vns7rcCtx7tfsxstbs3jkFI44bOORp0ztEQ1jmHOY5sJzD4rpvZwbphy5hZAqgm22ksIiI5EmYiWAUsNrP5ZlYMXAGsGFJmBXBN8P4y4DeF1D8gIhIFoTUNuXvKzD4K/JLs8NE73H29mX0JWO3uK4Dbge+b2WaglWyyCNNRNy+NQzrnaNA5R0Mo5xxaZ7GIiIwPutdcRCTilAhERCIuMonAzC40s01mttnMPpvveMJgZg1m9oiZPW9m683s48H6WjN7yMxeDH5OqMevmVnczP7HzH4RLM83s5XBtb4nGKwwYZhZjZndZ2YbzWyDmZ0ZgWv8yeBvep2Z3W1mpRPtOpvZHWbWHNxfNbBu2OtqWTcH5/6smZ1yNMeORCIYNN3FRcCxwPvM7Nj8RhWKFHCjux8LLAc+EpznZ4Ffu/ti4NfB8kTycWDwLcz/AnzN3RcB+8lOZTKR/AfwoLsvA04ke+4T9hqb2SzgY0Cjux9HdvDJwJQ0E+k6fxe4cMi6g13Xi4DFwesG4JtHc+BIJAIGTXfh7v3AwHQXE4q773L3p4P3HWQ/IGaRPdc7g2J3Au/IT4Rjz8xmA38G3BYsG3A+2SlLYOKdbzXwRrIj7nD3fndvYwJf40ACKAvuNyoHdjHBrrO7P0p29ORgB7uulwLf86wngRozm3Gkx45KIpgF7Bi03BSsm7DMbB5wMrASmObuu4JNu4Fwn6SeW/8O/DUw8BzIOqDN3VPB8kS71vOBFuA7QXPYbWZWwQS+xu6+E/g3YDvZBHAAWMPEvs4DDnZdx/QzLSqJIFLMrBL4CfAJd28fvC24YW9CjBk2s7cBze6+Jt+x5FACOAX4prufDHQxpBloIl1jgKBd/FKySXAmUMHrm1AmvDCva1QSwWimu5gQzKyIbBK4y91/GqzeM1BtDH425yu+MXY28HYz20a2ue98su3nNUETAky8a90ENLn7ymD5PrKJYaJeY4A3Ay+5e4u7J4Gfkr32E/k6DzjYdR3Tz7SoJILRTHcx7gXt47cDG9z9pkGbBk/lcQ3ws1zHFgZ3/5y7z3b3eWSv6W/c/UrgEf44f/SEOV8Ad98N7DCzgfmuLyA7tfuEvMaB7cByMysP/sYHznnCXudBDnZdVwBXB6OHlgMHBjUhHT53j8QLuBh4AdgC/G2+4wnpHM8hW3V8FngmeF1Mtt3818CLwMNAbb5jDeHczwN+EbxfADwFbAZ+DJTkO74xPteTgNXBdb4fmDzRrzHwRWAjsA74PlAy0a4zcDfZPpAk2ZrfdQe7roCRHQm5BXiO7IiqIz62ppgQEYm4qDQNiYjIQSgRiIhEnBKBiEjEKRGIiEScEoGISMQpEYgcBjNzM/vBoOWEmbUMmvn03cEsmY+ZWV2wbqGZ3ZOvmEVGokQgcni6gOPMrCxYfguvvaPzr4DTgG8B7w/W/SPwdzmLUOQwKRGIHL4HyM54CvA+sjcCDciQvdmpHEia2bnAbnd/MbchioyeEoHI4fsRcIWZlQInkJ3hdcA/k70D9BKyCeLvgS/nPEKRw6BEIHKY3P1ZYB7Z2sADQ7Y95O6nuvslZGfMfABYEjxR7NtmVp7zgEVGoEQgcmRWkJ0j/+7hNgYf+NeSnQ/mi2QnDPs9cGWO4hMZtcTIRURkGHeQfTDKc2Z23jDbPwPc7O7JoGPZyfYfqEYgBUeJQOQIuHsTcPNw28xsJnC6u38xWPWfZKdCb2OcP05RJibNPioiEnHqIxARiTglAhGRiFMiEBGJOCUCEZGIUyIQEYk4JQIRkYhTIhARibj/D6Hl3KPJ3dz+AAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "a = sns.histplot(data=ICM, kde=True, x=\"M%\", bins=10)\n",
        "a.set_title(\"ICM\")\n",
        "a.set(ylim=(0,3500000))\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "aZ0IrLVmt5ol",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 295
        },
        "outputId": "a6fc42b3-7c01-4d23-fcda-b937bcb14526"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZxddX3/8dfnrrMmk0wmk22SkI1IgLAkGAERQWRRSVVkKRWpC9bWrVRbbV0qra3212JVqEoRFayIINKICIJSAYWQhCVkAxLIMkOWyWSZzH7nzuf3xz0TbiYzmZlkztxkzvv54D5y7zln7vlcTnLf8/1+z/kec3dERCS6YoUuQERECktBICIScQoCEZGIUxCIiEScgkBEJOIUBCIiEacgEBGJOAWByACY2UYze1vwfKKZfd/MtprZPjNbZ2ZfMbPSYL2b2Q4zS+T9fDJYpgt35KijIBAZBDMbCzwJFANvcvdy4AKgApiZt+lu4OK81xcHy0SOOgoCkcG5HtgH/Jm7bwRw9y3u/il3X5m33R3ANXmvrwFuH7YqRQZBQSAyOG8D7nX3rn62uw84x8wqzGwM8Gbgf0OvTuQwHJNBYGa3Bf2tqwa4/eVmtsbMVpvZT8KuT0a0SmDrALZrA34JXBE8lgTLRI46if43OSr9ELiJATS1zWw28HngLHffbWbjQ65NRrYGYOIAt70d+FfAgL8LrSKRI3RMtgjc/TFgV/4yM5tpZg+a2Qoze9zM5garPgLc7O67g5/dMczlysjyCPBuMxvIv53HyYVGNfBEqFWJHIFjMgj6cAvwCXc/HfgM8F/B8jnAHDP7g5k9ZWYXFaxCGQluBEYBPzKzaQBmNtnMbjSzk/M39Nwc7+8CLnXN9y5HsWO1a+gAZlYGnAncbWbdi9PBnwlgNnAuMAV4zMxOcvc9w12nHPvcfZeZnQn8M7A0uHagDrgTWN/L9quHuUSRQRsRQUCuZbPH3U/pZV0tsNTdM8CrZvYSuWBYNpwFyrHN3afnPX8N+OAhtrU+lq8nN14gclQZEV1D7t5I7kv+fQCWMz9YfR+51gBmNo5cV9ErhahTRORoFFoQmFmRmT1tZs8Hp21+pZdtrjWzejN7Lnh8eIDvfSe5qzuPN7NaM/sQcDXwITN7HlgNLA42fwhoMLM1wKPAZ929YSg+o4jISGBhjWFZrrO+1N2bzCxJ7qyJT7n7U3nbXAsscPePh1KEiIj0K7QxguAsiabgZTJ46MwJEZGjTKiDxWYWB1YAs8idy7+0l83ea2bnAC8Bf+3uW3p5n+uA6wBKS0tPnzt3bs9NRETkEFasWLHT3at6Wxda19ABOzGrAH5B7jz/VXnLK4Emd283s48CV7j7eYd6rwULFvjy5cvDLVhEZIQxsxXuvqC3dcNy1lBwzv6jwEU9lje4e3vw8lbg9OGoR0REXhfmWUNVQUsAMysmN2f7uh7b5M/ZcimwNqx6RESkd2GOEUwkdxl+nFzg/Mzd7zezG4Dl7r4E+KSZXQp0kps76NoQ6xERkV4MyxjBUNIYgYjI4BV8jEBERI5eCgIRkYhTEIiIRJyCQEQk4hQEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGIUxCIiERcZILgjxt28r7v/pEd+9oKXYqIyFElMkHQ1QXLNu5m/Y6m/jcWEYmQyATBzPGlAGyoby5wJSIiR5fIBMGEUUWUpOJsUItAROQAkQkCM2NmVRkb6hUEIiL5IhMEALPGl/GKuoZERA4QqSCYWVVK3Z5Wmts7C12KiMhRI2JBUAbAqzvVKhAR6RatIBifCwKNE4iIvC5SQTCtsoSYoTOHRETyRCoI0ok4U8eW6FoCEZE8oQWBmRWZ2dNm9ryZrTazr/SyTdrM7jKz9Wa21Mymh1VPt3FlaXY1d4S9GxGRY0aYLYJ24Dx3nw+cAlxkZot6bPMhYLe7zwK+AXw9xHoAKC9KsK89E/ZuRESOGaEFged0d8Yng4f32Gwx8KPg+T3A+WZmYdUEUF6UpKlNp4+KiHQLdYzAzOJm9hywA3jY3Zf22GQysAXA3TuBvUBlL+9znZktN7Pl9fX1R1RTeVGCfQoCEZH9Qg0Cd8+6+ynAFOAMMzvxMN/nFndf4O4LqqqqjqimMgWBiMgBhuWsIXffAzwKXNRjVR1QA2BmCWA00BBmLaOKknRku2jLZMPcjYjIMSPMs4aqzKwieF4MXACs67HZEuADwfPLgN+5e89xhCFVXpQAUKtARCSQCPG9JwI/MrM4ucD5mbvfb2Y3AMvdfQnwfeAOM1sP7AKuDLEe4PUgaGrvpKo8HfbuRESOeqEFgbuvBE7tZfmX8p63Ae8Lq4belKeTAOxr0ymkIiIQsSuLITdYDOoaEhHpFrkgeH2MQC0CERGIYBCMKsp1DTWqRSAiAkQwCPYPFisIRESACAZBWVpjBCIi+SIXBIl4jOJkXGMEIiKByAUBaL4hEZF80Q0CTUUtIgJENgiSahGIiAQiGgTqGhIR6RbhIFDXkIgIRDUI0uoaEhHpFs0gUNeQiMh+EQ2CJK2ZLJlsV6FLEREpuIgGQe7q4uZ2tQpERCIZBJqKWkTkdZEMglFBEDTqzCERkWgGQVlwlzLNQCoiEtEgKE7FAWjJZAtciYhI4UUyCEqCIGjtUBCIiEQ6CFoUBCIi0QyC4v0tAo0RiIiEFgRmVmNmj5rZGjNbbWaf6mWbc81sr5k9Fzy+FFY9+UpSubOG1CIQEYFEiO/dCfyNuz9jZuXACjN72N3X9NjucXd/Z4h1HKQ4qa4hEZFuobUI3H2ruz8TPN8HrAUmh7W/wYjHjHQiRqvOGhIRGZ4xAjObDpwKLO1l9ZvM7Hkz+7WZzevj568zs+Vmtry+vn5IaipJxWnRGIGISPhBYGZlwM+BT7t7Y4/VzwDT3H0+8G3gvt7ew91vcfcF7r6gqqpqSOoqSSXUNSQiQshBYGZJciHwP+5+b8/17t7o7k3B8weApJmNC7OmbsWpuK4jEBEh3LOGDPg+sNbdb+xjmwnBdpjZGUE9DWHVlK80FVeLQESEcM8aOgt4P/CCmT0XLPt7YCqAu38XuAz4mJl1Aq3Ale7uIda0n1oEIiI5oQWBuz8BWD/b3ATcFFYNh1KSSrBjX1shdi0iclSJ5JXFkGsRqGtIRCTCQVCSjNPSriAQEYluEOg6AhERIMJBUJxK6MpiEREiHAQlqTiZrJPJdhW6FBGRgop0EIAmnhMRiWwQFOsuZSIiQISD4PUWgQaMRSTaIhsExUndnEZEBCIcBPtvYK8zh0Qk4iIfBGoRiEjURTYIdAN7EZGcyAaBbmAvIpIT4SBQ15CICCgIdB2BiERehINAXUMiIhDhIIjHjFQiRktGg8UiEm2RDQLIdQ+pa0hEoi7aQZCM06yb04hIxEU6CIpTcVrVNSQiERfpIChJJTRYLCKRF+kg0A3sRURCDAIzqzGzR81sjZmtNrNP9bKNmdm3zGy9ma00s9PCqqc3GiwWEQm3RdAJ/I27nwAsAv7KzE7osc3FwOzgcR3wnRDrOYhuYC8iEmIQuPtWd38meL4PWAtM7rHZYuB2z3kKqDCziWHV1FNxMqEWgYhE3rCMEZjZdOBUYGmPVZOBLXmvazk4LEJTkorTovsRiEjEhR4EZlYG/Bz4tLs3HuZ7XGdmy81seX19/ZDVVqLBYhGRcIPAzJLkQuB/3P3eXjapA2ryXk8Jlh3A3W9x9wXuvqCqqmrI6itOxeno7CLb5UP2niIix5owzxoy4PvAWne/sY/NlgDXBGcPLQL2uvvWsGrqSTewFxGBxEA2MrOz3P0P/S3r4Szg/cALZvZcsOzvgakA7v5d4AHgEmA90AL8+eDKPzLFwQykrR1ZyouSw7lrEZGjxoCCAPg20PMc/96W7efuTwB2qDd1dwf+aoA1DLmSpG5OIyJyyCAwszcBZwJVZnZ93qpRQDzMwoZDaVpBICLSX4sgBZQF25XnLW8ELgurqOGyv2tIE8+JSIQdMgjc/ffA783sh+6+aZhqGja6b7GIyMDHCNJmdgswPf9n3P28MIoaLsUaIxARGXAQ3A18F7gVGDHfmjp9VERk4EHQ6e7DOiHccNAN7EVEBn5B2S/N7C/NbKKZje1+hFrZMCgOWgSaeE5EomygLYIPBH9+Nm+ZAzOGtpzhpcFiEZEBBoG7Hxd2IYWQjMdIxk1BICKRNtApJq7pbbm73z605Qy/4mScVg0Wi0iEDbRraGHe8yLgfOAZ4JgPAt3AXkSibqBdQ5/If21mFcBPQ6lomOnmNCISdYc7DXUzMCLGDYp1A3sRibiBjhH8ktxZQpCbbO4NwM/CKmo46Qb2IhJ1Ax0j+Pe8553AJnevDaGeYVecSrC3paPQZYiIFMyAuoaCyefWkZuBdAwwYr45S5K6b7GIRNuAgsDMLgeeBt4HXA4sNbNjfhpq0A3sRUQG2jX0D8BCd98BYGZVwCPAPWEVNlxK0nFaddaQiETYQM8ainWHQKBhED97VMtdR6DBYhGJroG2CB40s4eAO4PXV5C78fwxrzgZpy3TRVeXE4sd8hbLIiIjUn/3LJ4FVLv7Z83sPcDZwaongf8Ju7jh0D3xXGsmS2l6oLkoIjJy9Ne985/k7k+Mu9/r7te7+/XAL4J1x7zuIGhW95CIRFR/QVDt7i/0XBgsm36oHzSz28xsh5mt6mP9uWa218yeCx5fGnDVQ2j/Dex15pCIRFR/fSEVh1hX3M/P/hC4iUNPTPe4u7+zn/cJle5JICJR11+LYLmZfaTnQjP7MLDiUD/o7o8Bu46gtmFRrCAQkYjrr0XwaeAXZnY1r3/xLwBSwLuHYP9vMrPngdeAz7j76iF4z0EpSep2lSISbYcMAnffDpxpZm8FTgwW/8rdfzcE+34GmObuTWZ2CXAfMLu3Dc3sOuA6gKlTpw7Brl/3+g3sNVgsItE00PsRPAo8OpQ7dvfGvOcPmNl/mdk4d9/Zy7a3ALcALFiwwHuuPxLFeaePiohEUcGuDjazCWZmwfMzgloahrsODRaLSNSFdgWVmd0JnAuMM7Na4MtAEsDdvwtcBnzMzDqBVuBKdx/S3/YHQkEgIlEXWhC4+1X9rL+J3OmlBbW/a0hjBCISUSNi4rgjkYrHiMdMLQIRiazIB4GZ6eY0IhJpkQ8C0A3sRSTaFAQEdynT6aMiElEKAnIXlWmwWESiSkGA7lssItGmICA3RqAgEJGoUhDQ3SJQ15CIRFOkgqBm6jTM7KDHz++6kzUvru913VA9aqZOK/THFxHpVaRu0lu7ZTM3/ubFg5b/dt12Nuxo7nXdULn+7ceH9t4iIkciUi2CviTjMTLZrkKXISJSEAoCIBmL0dnlFGDOOxGRglMQAMm4AdDZpSAQkehREJDrGgLUPSQikaQgABJBiyCTVYtARKJHQYBaBCISbQoCXg+CTrUIRCSCFAS8PlisFoGIRJGCAEioa0hEIkxBACRjGiwWkehSEJA3WNylFoGIRI+CAEgmNFgsItGlICC/a0gtAhGJntCCwMxuM7MdZraqj/VmZt8ys/VmttLMTgurlv7EY4ahIBCRaAqzRfBD4KJDrL8YmB08rgO+E2Ith2RmwQyk6hoSkegJLQjc/TFg1yE2WQzc7jlPARVmNjGsevqTiJtaBCISSYUcI5gMbMl7XRssO4iZXWdmy81seX19fSjF6J4EIhJVx8Rgsbvf4u4L3H1BVVVVKPtIxk1nDYlIJBUyCOqAmrzXU4JlBaEWgYhEVSGDYAlwTXD20CJgr7tvLVQxuTECtQhEJHpCu3m9md0JnAuMM7Na4MtAEsDdvws8AFwCrAdagD8Pq5aBSMVjtLRnClmCiEhBhBYE7n5VP+sd+Kuw9j9YqUSM9k51DYlI9BwTg8XDIZ2I06EgEJEIUhAE0okYHdkuunQDexGJGAVBIB1MPNeuM4dEJGIUBIF0Mg5AeyZb4EpERIaXgiBQ1N0i0DiBiESMgiCQTgQtAgWBiESMgiCQ2t8iUNeQiESLgiCQTgZBkFGLQESiRUEQKFLXkIhElIIgkIzn7lKmriERiRoFQcDMSGuaCRGJIAVBnnQyriAQkchREORJJ2K6oExEIkdBkEddQyISRQqCPOmEuoZEJHoUBHnSyZjOGhKRyFEQ5MmNEahFICLRoiDIk07E6exysrongYhEiIIgT1rzDYlIBCkI8uyfb0gDxiISIQqCPPunotY4gYhEiIIgj7qGRCSKQg0CM7vIzF40s/Vm9rle1l9rZvVm9lzw+HCY9fSnOwja1CIQkQhJhPXGZhYHbgYuAGqBZWa2xN3X9Nj0Lnf/eFh1DEZJKve/o6Wjs8CViIgMnzBbBGcA6939FXfvAH4KLA5xf0esKBkjZtDcoa4hEYmOMINgMrAl73VtsKyn95rZSjO7x8xqensjM7vOzJab2fL6+vowau3eDyWpBC3tahGISHQUerD4l8B0dz8ZeBj4UW8bufst7r7A3RdUVVWFWlBpOq4WgYhESphBUAfk/4Y/JVi2n7s3uHt78PJW4PQQ6xmQ0lSCZrUIRCRCwgyCZcBsMzvOzFLAlcCS/A3MbGLey0uBtSHWMyCl6QQtahHIIG3d28oTL+9k/Y59hS5FZNBCO2vI3TvN7OPAQ0AcuM3dV5vZDcByd18CfNLMLgU6gV3AtWHVM1AlqTitmSzZLices0KXI8eArXtbueDGx2gKWpI/+fAbOXPWuAJXJTJwoY4RuPsD7j7H3We6+1eDZV8KQgB3/7y7z3P3+e7+VndfF2Y9A1Ga1imkMjhfWbKGzq4ufnDtQqZVlvCF+1bpokQ5phR6sPioU5rKTTPR3K5/yNK/x16q58HV2/jk+bN569zx3LD4RF7Z2cytj79a6NJEBkxB0EN3i6BZLQIZgLuWbWFcWYoPnz0DgLfMqeLNs8fxk6Wb6dJ05nKMUBD0UBpcXawzh6Q/rR1ZfrduBxfOm0Aq8fo/pfeeNoW6Pa0s27irgNWJDJyCoIeSoGtIZw5Jf/7vxR20ZrK846SJByx/+7xqSlJx7nvutQJVJjI4CoIeYjGjOBlXi2AY/Hbtdq743pN8/t6VrKrbW+hyBu2BVdsYW5rijOPGHrC8JJXgwnkT+NXK1zRoLEPG3XEPp7tRQdALXV0cvgdXbeWjd6ygdncr9z+/lQ/c9jQ7GtsKXdaAtWWy/Hbtdi6cV00ifvA/o0tPmURjWyd/WL+zANXJSPPqzmY+8INl/OLZuv43PgwKgl7o6uJw7Whs49N3PcfJU0bz4KffzL1/eSbNHZ389c+eO2YGWH//Uj0tHVku6dEt1O3MmZWUpRM8tGr7MFcmI82vVm7lwm88xrObdtOZVYtg2JQVJdjXpiAIy7d/t57OrPONK06hvCjJ7OpyvvTOefxhfQO/XrWt0OUNyK9f2EpFSZJFMyp7XZ9OxHnr3PE8snY72WMk3OTos7u5gy/c9wJvmFjObz/zFi5f2Ou8nEdMQdCLsSUpWjNZWtU9NOQ2N7Rw59ObuWJhDdMqS/cvv2JhDceNK+U7v18fWj/oUGnLZHlk7Q4uPGECyV66hbpdOK+ahuYOluvsITlMX/v1OhrbOvn6ZSczvrwotP0oCHpRWZYCoKG5vZ8tZbC++9gGYjHjk+fPPmB5PGZ89JwZrKpr5ImjvF/9iZd30tTeycUnTTjkduceP55UIsZDq9U9JIO3ob6Ju5Zv4c/PnM7cCaNC3ZeCoBdjS7uDoKPAlRROtsvZ1NA8pFNt7Gru4N5nannPqZOpHnXwbzfvPm0y1aPSfO/3rwzZPsPwwAtbGVWU4MyZh55PqCyd4OxZ43ho9bajvpUjR58f/OFVUvEYH33LzND3pSDoRVk6QSoeY1dT9ILA3bnlsQ2c+OWHeMv/+z/O/vqj3PHkxiH5IvvJ0k20Zbr44NnH9bo+nYjz/kXTeGL90TuLZ3tnlofXbuftPS4i68uF86qp29PK6tcah6E6GSn2tHTw8xV1XHrKJKrK06HvT0HQCzOjsiwVuRZBV5fzt/es5F8eWMdZsyr56rtPZE51GV/839X8x29eOqL37ujs4vYnN3HOnCrmVJf3ud2VZ0wlFY9xx5Objmh/Yfnj+gb2tXVyST/dQt3e9oZqYga/WX1sDILL0eHOp7fQmsnywbN6/6VpqCkI+lBZmqKhuT1STfr/ePhF7l5RyyfOm8V/X7OAq984jTs/soirzqjhpkfX85Olmw/7ve9f+Ro79rXzoT5aA93GlaV558kTuWdFLfvaMoe9v7D86oWtlKcTnDXAaaYry9IsmD6W36zROIEMTCbbxY/+uJEzZ1ZywqRwxwa6KQj6MLY0RVumKzJTTdz7TC03P7qBKxfWcP0FczDL3YvBzPinxSdy7vFV/OOS1Yd1BbC7c+vjrzJ7fBnnzO7/C/Tas6bT3JHlp09v6Xfb4dTemeXhNdu54IRq0ol4r9vUTJ2GmR3wePDWr7Fu2z6SYycftG6oHjVTpw3z/w0Jy69XbWNbY1u/vzQNpdBuTHOsqyzL9cvtau7YPyPpsaSpvZNte9uIGUyrLD3kTXaWbdzF537+AotmjOWGxSfuD4FuiXiMGy8/hYu/+RifvPNZfvmJswf1/+SpV3axZmsjX3vPSQe9d29OnlLBmTMrufWJV7jmzGl9fukOtwdXbWNva4bFp07uc5vaLZu58TcvHrCsqa2T2/74Kpd+9W7ODumGNde//fhQ3leGl7vz/Sde5bhxpbz1+PHDtt9j7xtumIwLTiHdureNmrElBa5mYNoyWX75/GvcvaKWZzbtpjO4kCnb2kjry0tpXPFLMjsOPCMnPfkEqt77Bbpa93H3jZdz10eb+nz/dM1JVF/1Vaa95zM0PPDNfuuJJ5JkOzNUX/1vJCsm8qdnzeaqzoGNuxRNm0/1lV9l3OmX0LTyNwP6mW5TaqayZfPQjzH8+KlNTKss4c2D/DIvK0owY1wpa15rZNGMsSRiR2dDPNvl/OLZOu5f+Rq7mjuYO6Gcq984jfk1FYUuLTKefKWB57fs4Z8WzyM2jHdIVBD0oSSVYMKoItbXNx00qdjRxN1ZVdfI/z5Xx90ratnbmmFmVSkfOWdG7h/yNddy9ge/yIayt1N28gVMrihm3qRRlKUTbN7VwrOb91BelGDxKSdR8b4V/e7vyQ0NPG0X8N4r/6zfc5uvf/vx/MUdy/n1qm2cP3c8Jy5+YVCf66fLtlD2J9fz/q99c0Bn6OTvd6it29bIso27+fzFcw/rH+hJk0ezob6ZDTuaOX5C34PlhVK3p5WP/Gg5a7Y2MmNcKZMqirl/5VZ+tryW686ZwWcvPP6QF89FSc3UadRuOfzxskOpvupfSYyZxDVvnsM12YPHyML6JUdBcAizq8t4/OWd7GnpoKIkVehyAGhsy7Bi426e27KHF+r2srJ2DzubOkjEjAvnTeDPFk1j0Yyx+7tgml94hAvn3Ux7Jsuq1xp5vnbP/oFLM5gxrpS3vaGaouTAul/eeNxYane38MjaHZSnk0weU9zntrF0KU+s38m4stSgB73MjLfMqeLuFbU89UoD58ypGtTPD7VbHnuFVCLG+xYc3iX+U8eWMLo4yTObdzOnumxAXWTDZVXdXj74w2W0ZrLc9Ken8o6TJmJmNLV38m8PruOWx17hhdq93HLN6ZQXJQtdbsH11v03JO+7u4WfP1PHW+ZUccqvV/W6TVhdgAqCQ5g1PhcEL+9oYuH0wrQKduxrY9mru1m2cRdLX93Fum2NuEPMcvW9Zc543njcWC44oZoxpX2HVToZ5/RpYzi1poLdLR00tXcyriw96PGPWMx458mTuHvFFpY8/xrvmj+RKWMO7jrLdjlV7/57WtqzXHziBGKH8cU3qaKYkyaP5rkte5hWWXLAlBT52juzvLy9iU0NLexsbmfyx27jPf/1B+bXVPDe06Zw4uTRg953vhWbdnHvM3V87NyZ+y82HCwzY9FxY3lozXbWbG1k3qQjq2mo/P6lev7yxysYXZzk5x8784BTe8vSCW5YfCKn1FTwt/es5IrvPcUPP7gw1KkOoirb5fz+pXpK03FOHKYzhfIpCA5hVFGSCaOKeGn7PhZMGxP6b3HuzqaGFpZt3MXTr+5i2cZdbGxoAaA4Gee0aRV86vzZnDF9LPNrKg5rEDsWMyrL0vsHww9HcSrOn5w6mfuerePeZ+o4fdoYTsmrZ3dLB79bt4OiafN52xvGM3F0362G/pw1q5LX9rbyy5VbufCEamaNz/027e5s39fO6rq9vLh9H5msU16UoHpUEbWbVpKYP5efLN3MD/6wkZMmj+bKM2q4dP6kQf9G25bJ8sX7VjNxdBGfOG/WYX8OgOMnlLOybi9/WN/AjKoyigfYCguDu3P7k5u44f41HF9dzg/+fGGvV3sDvOe0KYwtTfGxHz/DZd95kts/eAbTx/UeynJ4nt64i51NHbzz5Im9TmseNgVBP06eMprfrNnOik27WTDErYLGtgyr6vby7OY9PLt5N89u3rP/IrbRxUkWTh/Ln75xKgunj+XEyaOPqj7aUUVJrlw4lUdf3MHyTbt5ZvNuRhUnwWFPa4ZUPMbOB/6TuefffET7SSfivPe0Kdz3bB0PrNrG6OIkZekEe1o7aG7PkogZc6rLOWnKaKrL05gZv/2b/+Rnv/oGe1sy3PdcHXc+vZl/+MUq/vn+tbzj5IlcvqCG06ZW9PsPri2T5aN3rGDN1ka+9/7TKUkd2T8XM+Otx4/nrmVbuO/ZOt5z6mTShxkG7Zksu1o6aGjuYF9rJ6PPuory+W+nY8dGMjs34519z5MVLx3DmPM+TOkJb6Hl5aU8+B//zoRPt/a7z9TEOTRd9mXO+ef7aXjoZlrWPT6oPuvm9k6Wb9rN0682sLGhhcbWDEXJ+P5xq0UzKo+ZEzOG0ob6JpZt3MXcCeXMrCorSA2hBoGZXQR8E4gDt7r713qsTwO3A6cDDcAV7r4xzJoGa+6Ecl7d2cwfX2mgvCg5qP5dd6cj20VTWydFx51G+fwLSY6buv+RKH/97JNMwxbaX3sx96hdw6adm1mJ8/2wPtgQSCViXDhvAmccN5a1WxvZ3ZLB3XnDpFGcMHEUX/qXR4ZkP8XJOO9bMGlJ4U8AAAnaSURBVIX125tYt30f2awzaXQx0ypLmFVVdvCXqcUOOkapiXMoO/nt/KzpHO5ZUUtXewtttatp37KazK46Ovdso7NxB2Q7iZdWkK6Zx+hF7yMxdjK7Hvw2F3394SH5LFXlaS45eQK/WrmVu5Zv4axZ45jex+m92S6nqb2TxtYMu1s62NX8+qPnjZMqzr76gNdjSpKMLy9ifHmaipIk8ZjR3JFly64W1u9owh0WHjeGM867GvuLPxtw/XtbMzy4ahuxxX/HpGu+zHM/+CLN7Z29tk4b2zKs2LSbpa/sYumrDbxQu5fOLicRM2rGllBRkqR+XztPvLyT1kzu80yuKOZNMyt504xK5teMZsqYEtKJg4/nSNDlzuq6Rh59aQfV5UWce3zhxsFCCwIziwM3AxcAtcAyM1vi7mvyNvsQsNvdZ5nZlcDXgSvCqulwmBnnzx3PnpYMD67extMbU0wYVUQ6GaP7r2Znl9PR2UVHZxftwZ8d2S5aOjrJBDeSqL78BiA3y2ZlaYqxpSkqS1OMK0szYXQRRcnZwHlDXv9wnF8+piTV7wRsRyoRizF34ijmThxA/6l39TmY19HZxcaGZmp3t1I7ZjS7Zy7s820qS1OcOauSGW+7acB1DuT/94xxZVw6fxL/91I996/cSiJmjClJkUwY2by/Sy0dWfKva0/GjbGlKaZWljB2/9+hNOVFCT5z0Ql8+b6V7Gxqp35f7lG3p5UXtx84Z1MqHuP4CeWcPm0MYw7jBIjRxUkuO30Kq+r2snzTbsa/5x844Qu/onPvdrLNe6Ari6VLiJeOIVGeu1eDZztp3/Yy7ZtfoG3zStrr1rIhk99iMZLjakjXnETztJPZvPUk7llx4HH2zgzelQUDsxgEj2QiTkkqTkkqQUk673kqTmkqQXEqTmkqTnGwrHt9aTpOcTJOaTpBKhEjZkYiZsRjlnsez/0Zj+WWx2JGPHgdKx5FWyZLfjYZr784cPnrsu60Z7rY19bJa3tbeWn7PnY2dTBlTDHvOnnSoM6MG2phtgjOANa7+ysAZvZTYDGQHwSLgX8Mnt8D3GRm5kfZvA7pZJwrF9awdlsj67btY2NDM5lsF+7gQCJmpBIx0okYqUSMsqLcX67iZJyydIKydIIf/v0H+NzNd1FelDisgVMZGqlEjDnV5fsHRdsyWfa2ZmhszbCvrZMunKJEnHFlaapHpUP7TXRaZSnvf2MJr+xspm5PK3taOshknaJkjFFFSVKJGKWpBOXFCUYVJakoSVKeTvRdj3cxujjJ6OLkAd0LLR2dNLZ1ks06pek45UXJQ15cOBDxmDG/poITJ4/mi9ddzjs//x32tFTQmsnS5U4qHqMklWB0SW6MbeLoIpLxNwCXDuj93Z2dTR00NLXT2J6rvctzD8MI/uORO7+LWYxdyTSxVBGWzD1iqWIsmQ6eB8uSxVhiaM54qvnkT/jeY0c+Q25VWZqL5k04Ks4is7C+c83sMuAid/9w8Pr9wBvd/eN526wKtqkNXm8IttnZ472uA64LXh4PHO65W+OAo3uy+6GnzxwN+szRcCSfeZq799r/dEwMFrv7LcAtR/o+Zrbc3RcMQUnHDH3maNBnjoawPnOYnVJ1QP7VN1OCZb1uY2YJYDS5QWMRERkmYQbBMmC2mR1nZingSmBJj22WAB8Inl8G/O5oGx8QERnpQusacvdOM/s48BC500dvc/fVZnYDsNzdlwDfB+4ws/XALnJhEaYj7l46BukzR4M+czSE8plDGywWEZFjw9FzqaqIiBSEgkBEJOIiEwRmdpGZvWhm683sc4WuJwxmVmNmj5rZGjNbbWafCpaPNbOHzezl4M8xha51KJlZ3MyeNbP7g9fHmdnS4FjfFZysMGKYWYWZ3WNm68xsrZm9KQLH+K+Dv9OrzOxOMysaacfZzG4zsx3B9VXdy3o9rpbzreCzrzSz045k35EIgrzpLi4GTgCuMrMTCltVKDqBv3H3E4BFwF8Fn/NzwG/dfTbw2+D1SPIpYG3e668D33D3WcBuclOZjCTfBB5097nAfHKffcQeYzObDHwSWODuJ5I7+aR7SpqRdJx/CFzUY1lfx/ViYHbwuA74zpHsOBJBQN50F+7eAXRPdzGiuPtWd38meL6P3BfEZHKf9UfBZj8C/qQwFQ49M5sCvAO4NXht5CZtuifYZKR93tHAOeTOuMPdO9x9DyP4GAcSQHFwvVEJsJURdpzd/TFyZ0/m6+u4LgZu95yngAozm3i4+45KEEwGtuS9rg2WjVhmNh04FVgKVLv71mDVNqC6QGWF4T+BvwW6gteVwB537wxej7RjfRxQD/wg6A671cxKGcHH2N3rgH8HNpMLgL3ACkb2ce7W13Ed0u+0qARBpJhZGfBz4NPu3pi/Lrhgb0ScM2xm7wR2uHv/N1seORLAacB33P1UoJke3UAj6RgDBP3ii8mF4CSglIO7UEa8MI9rVIJgINNdjAhmliQXAv/j7vcGi7d3NxuDP3cUqr4hdhZwqZltJNfddx65/vOKoAsBRt6xrgVq3X1p8PoecsEwUo8xwNuAV9293t0zwL3kjv1IPs7d+jquQ/qdFpUgGMh0F8e8oH/8+8Bad78xb1X+VB4fAP53uGsLg7t/3t2nuPt0csf0d+5+NfAouSlLYAR9XgB33wZsMbPuGx+cT25q9xF5jAObgUVmVhL8He/+zCP2OOfp67guAa4Jzh5aBOzN60IaPHePxAO4BHgJ2AD8Q6HrCekznk2u6bgSeC54XEKu3/y3wMvAI8DYQtcawmc/F7g/eD4DeBpYD9wNpAtd3xB/1lOA5cFxvg8YM9KPMfAVYB2wCrgDSI+04wzcSW4MJEOu5fehvo4ruVsy3Bx8n71A7oyqw963ppgQEYm4qHQNiYhIHxQEIiIRpyAQEYk4BYGISMQpCEREIk5BIDIIZuZm9uO81wkzq8+b+fS9wSyZj5tZZbBsppndVaiaRfqjIBAZnGbgRDMrDl5fwIFXdH4CWAh8D/jTYNk/A18YtgpFBklBIDJ4D5Cb8RTgKnIXAnXrInexUwmQMbM3A9vc/eXhLVFk4BQEIoP3U+BKMysCTiY3w2u3fyV3Bei7yAXEF4F/GvYKRQZBQSAySO6+EphOrjXwQI91D7v76e7+LnIzZj4AzAnuKPbfZlYy7AWL9ENBIHJ4lpCbI//O3lYGX/jXkpsP5ivkJgx7Arh6mOoTGbBE/5uISC9uI3djlBfM7Nxe1n8W+Ja7Z4KBZSc3fqAWgRx1FAQih8Hda4Fv9bbOzCYBZ7j7V4JF3yY3FfoejvHbKcrIpNlHRUQiTmMEIiIRpyAQEYk4BYGISMQpCEREIk5BICIScQoCEZGIUxCIiETc/we+VBnVDomRfQAAAABJRU5ErkJggg==\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JabQ0OLhLu7C",
        "outputId": "688f30d6-f4a6-477e-ab14-4901d866c1bf"
      },
      "execution_count": 36,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-13 19:10:12--  http://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1405 (1.4K)\n",
            "Saving to: ‘mm10.chrom.sizes’\n",
            "\n",
            "mm10.chrom.sizes    100%[===================>]   1.37K  --.-KB/s    in 0s      \n",
            "\n",
            "2023-02-13 19:10:12 (139 MB/s) - ‘mm10.chrom.sizes’ saved [1405/1405]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d s_8_cell.deduplicated.bedGraph.gz\n",
        "!gzip -d s_epiblast.deduplicated.bedGraph.gz\n",
        "!gzip -d s_ICM.deduplicated.bedGraph.gz"
      ],
      "metadata": {
        "id": "1DNs7NvHMQ-p"
      },
      "execution_count": 40,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
        "!sudo apt-get install /content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OWEZNKS1NT2r",
        "outputId": "23cd80b5-9044-49b7-de52-022d3c7dbeb9"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-13 19:16:51--  http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
            "Resolving archive.ubuntu.com (archive.ubuntu.com)... 185.125.190.39, 91.189.91.39, 185.125.190.36, ...\n",
            "Connecting to archive.ubuntu.com (archive.ubuntu.com)|185.125.190.39|:80... connected.\n",
            "HTTP request sent, awaiting response... 404 Not Found\n",
            "2023-02-13 19:16:51 ERROR 404: Not Found.\n",
            "\n",
            "Reading package lists... Done\n",
            "E: Unsupported file /content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb given on commandline\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda ucsc-bedgraphtobigwig\n",
        "\n",
        "! wget http://snapshot.debian.org/archive/debian/20190501T215844Z/pool/main/g/glibc/multiarch-support_2.28-10_amd64.deb\n",
        "! sudo dpkg -i multiarch-support*.deb\n",
        "\n",
        "! wget http://snapshot.debian.org/archive/debian/20170705T160707Z/pool/main/o/openssl/libssl1.0.0_1.0.2l-1%7Ebpo8%2B1_amd64.deb\n",
        "! sudo dpkg -i libssl1.0.0*.deb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SGNBiADKOOCE",
        "outputId": "b9868967-21c7-43f2-ab1e-d5815f148369"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n",
            "--2023-02-13 19:21:24--  http://snapshot.debian.org/archive/debian/20190501T215844Z/pool/main/g/glibc/multiarch-support_2.28-10_amd64.deb\n",
            "Resolving snapshot.debian.org (snapshot.debian.org)... 185.17.185.185, 193.62.202.27, 2001:630:206:4000:1a1a:0:c13e:ca1b, ...\n",
            "Connecting to snapshot.debian.org (snapshot.debian.org)|185.17.185.185|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 215080 (210K)\n",
            "Saving to: ‘multiarch-support_2.28-10_amd64.deb’\n",
            "\n",
            "multiarch-support_2 100%[===================>] 210.04K   472KB/s    in 0.4s    \n",
            "\n",
            "2023-02-13 19:21:25 (472 KB/s) - ‘multiarch-support_2.28-10_amd64.deb’ saved [215080/215080]\n",
            "\n",
            "Selecting previously unselected package multiarch-support.\n",
            "(Reading database ... 128110 files and directories currently installed.)\n",
            "Preparing to unpack multiarch-support_2.28-10_amd64.deb ...\n",
            "Unpacking multiarch-support (2.28-10) ...\n",
            "Setting up multiarch-support (2.28-10) ...\n",
            "--2023-02-13 19:21:27--  http://snapshot.debian.org/archive/debian/20170705T160707Z/pool/main/o/openssl/libssl1.0.0_1.0.2l-1%7Ebpo8%2B1_amd64.deb\n",
            "Resolving snapshot.debian.org (snapshot.debian.org)... 185.17.185.185, 193.62.202.27, 2001:630:206:4000:1a1a:0:c13e:ca1b, ...\n",
            "Connecting to snapshot.debian.org (snapshot.debian.org)|185.17.185.185|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1294986 (1.2M)\n",
            "Saving to: ‘libssl1.0.0_1.0.2l-1~bpo8+1_amd64.deb’\n",
            "\n",
            "libssl1.0.0_1.0.2l- 100%[===================>]   1.23M   926KB/s    in 1.4s    \n",
            "\n",
            "2023-02-13 19:21:29 (926 KB/s) - ‘libssl1.0.0_1.0.2l-1~bpo8+1_amd64.deb’ saved [1294986/1294986]\n",
            "\n",
            "Selecting previously unselected package libssl1.0.0:amd64.\n",
            "(Reading database ... 128114 files and directories currently installed.)\n",
            "Preparing to unpack libssl1.0.0_1.0.2l-1~bpo8+1_amd64.deb ...\n",
            "Unpacking libssl1.0.0:amd64 (1.0.2l-1~bpo8+1) ...\n",
            "Setting up libssl1.0.0:amd64 (1.0.2l-1~bpo8+1) ...\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_8_cell.deduplicated.bedGraph mm10.chrom.sizes cell8_methylation.bigWig"
      ],
      "metadata": {
        "id": "p7CjvnFhwpyL"
      },
      "execution_count": 56,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g mm10.chromSizes  >  s_8_cell.cover.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "d86SizYFaAIZ",
        "outputId": "24c7d9b9-a54e-4ecf-e563-e5968bb055dc"
      },
      "execution_count": 57,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig   s_8_cell.cover.bedGraph mm10.chrom.sizes cell8_coverage.bigWig"
      ],
      "metadata": {
        "id": "vyWxGTH2aNfu"
      },
      "execution_count": 58,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_epiblast.deduplicated.bedGraph mm10.chrom.sizes epiblast_methylation.bigWig\n",
        "! bedtools genomecov -bg -ibam s_epiblast.deduplicated.bam -g mm10.chromSizes  >  s_epiblast.cover.bedGraph\n",
        "! bedGraphToBigWig   s_epiblast.cover.bedGraph mm10.chrom.sizes epiblast_coverage.bigWig"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F0hERFnebFaQ",
        "outputId": "75813354-fb54-40c1-9846-c30e6c9c0642"
      },
      "execution_count": 59,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_ICM.deduplicated.bedGraph mm10.chrom.sizes ICM_methylation.bigWig\n",
        "! bedtools genomecov -bg -ibam s_ICM.deduplicated.bam -g mm10.chromSizes  >  s_ICM.cover.bedGraph\n",
        "! bedGraphToBigWig   s_ICM.cover.bedGraph mm10.chrom.sizes ICM_coverage.bigWig"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_BPvbiPdbdBk",
        "outputId": "13fa6071-ee40-4c42-b2f7-b0fe7afbb1ac"
      },
      "execution_count": 60,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! pyGenomeTracks --tracks tracks.ini.txt --region chr11:3100030-3500030 -o image_cov.png"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TmZOeAbydZ47",
        "outputId": "487c6c48-b0cc-4e41-d591-fbc45db2d502"
      },
      "execution_count": 64,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [cell8_coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [cell8_methylation]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [epiblast_coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [epiblast_methylation]\n",
            "INFO:pygenometracks.tracksClass:initialize 5. [ICM_coverage]\n",
            "INFO:pygenometracks.tracksClass:initialize 6. [ICM_methylation]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.003810882568359375\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 25.531914893617024. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [cell8_coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [cell8_methylation]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [epiblast_coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [epiblast_methylation]\n",
            "INFO:pygenometracks.tracksClass:plotting 5. [ICM_coverage]\n",
            "INFO:pygenometracks.tracksClass:plotting 6. [ICM_methylation]\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}